\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage[]{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\PassOptionsToPackage{hyphens}{url} % url is loaded by hyperref
\usepackage[unicode=true]{hyperref}
\hypersetup{
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage[margin=1in]{geometry}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother


\author{}
\date{\vspace{-2.5em}}

\begin{document}

\section{\texorpdfstring{Data and code for the paper: `Combining
climatic and genomic data improves range-wide tree height growth
prediction in a forest
tree'}{Data and code for the paper: Combining climatic and genomic data improves range-wide tree height growth prediction in a forest tree}}\label{data-and-code-for-the-paper-combining-climatic-and-genomic-data-improves-range-wide-tree-height-growth-prediction-in-a-forest-tree}

\textbf{Authors:} Juliette Archambeau\(^1\), Marta Benito Garzón\(^1\),
Frédéric Barraquand\(^2\), Marina de Miguel Vega\(^{1,3}\), Christophe
Plomion\(^1\) and Santiago C. González-Martínez\(^1\)

\textbf{1} INRAE, Univ. Bordeaux, BIOGECO, F-33610 Cestas, France

\textbf{2} CNRS, Institute of Mathematics of Bordeaux, F-33400 Talence,
France

\textbf{3} EGFV, Univ. Bordeaux, Bordeaux Sciences Agro, INRAE, ISVV,
F-33882, Villenave d'Ornon, France

\textbf{Published in \emph{The American Naturalist} } (manuscript \#
60477R2)

\subsection{Data}\label{data}

\subsubsection{Height and climatic data}\label{height-and-climatic-data}

\textbf{In the dataset
\texttt{HeightClimateSoilData\_33121obs\_25variables.csv}.}

The 33,121 height observations comes from 5 common gardens of the
CLONAPIN network, consisting of 34 populations (i.e.~provenances) and
523 clones (i.e.~genotypes).

Annual and monthly climatic data were extracted from the
\href{https://gentree.data.inra.fr/climate/}{EuMedClim database at 1-km
resolution}. The scripts for the extraction can be found here:
\href{https://github.com/JulietteArchambeau/HeightPinpinClonapin/blob/master/scripts/DataMunging/01_ExtractingMonthlyClimateEuMedClimData.R}{monthly
data for climate in the test sites} and
\href{https://github.com/JulietteArchambeau/HeightPinpinClonapin/blob/master/scripts/DataMunging/01_ExtractingAnnualClimateEuMedClimData.R}{annual
data for climate in the provenances}. We calculated six variables that
describe extreme and average temperature and precipitation conditions in
the test sites during the year preceding the measurements
(\href{https://github.com/JulietteArchambeau/HeightPinpinClonapin/blob/master/scripts/DataMunging/02_CalculatingSiteMonthlyClimaticVariables.R}{script
available here}) and four variables that describe the mean temperature
and precipitation in the provenance locations over the period from 1901
to 2009, representing the climate under which provenances have evolved
(\href{https://github.com/JulietteArchambeau/HeightPinpinClonapin/blob/master/scripts/DataMunging/02_CalculatingProvAnnualClimaticVariables.R}{script
available here}).

Missing data are indicated with \emph{NA}.

Meaning of the columns:

\begin{itemize}
\tightlist
\item
  \texttt{obs}: unique code differentiating each observation (i.e.~each
  height measurement)
\item
  \texttt{tree}: tree identity
\item
  \texttt{site}: test site (common garden)
\item
  \texttt{clon}: clone (i.e.~genotype)
\item
  \texttt{prov}: provenance (i.e.~population)
\item
  \texttt{latitude\_site}: latitude of the test site (in degrees)
\item
  \texttt{longitude\_site}: longitude of the test site (in degrees)
\item
  \texttt{latitude\_prov}: latitude of the provenance (in degrees)
\item
  \texttt{longitude\_prov}: longitude of the provenance (in degrees)
\item
  \texttt{age}: tree age when the measurement was taken (in months)
\item
  \texttt{height}: tree height (in mm)
\item
  \texttt{pre\_summer\_min\_site} (\emph{min.presummer} in the
  manuscript): minimum of the monthly precipitation during summer -June
  to September- in the test sites during the year preceding the
  measurement (°C)
\item
  \texttt{pre\_mean\_1y\_site} (\emph{mean.pre} in the manuscript): mean
  of the monthly precipitation in the test sites during the year
  preceding the measurement (mm)
\item
  \texttt{tmn\_min\_1y\_site} (\emph{min.tmn} in the manuscript):
  minimum of the monthly minimum temperatures in the test sites during
  the year preceding the measurement (°C)
\item
  \texttt{tmx\_max\_1y\_site} (\emph{max.tmx} in the manuscript):
  maximum of the monthly maximum temperatures in the test sites during
  the year preceding the measurement (°C)
\item
  \texttt{pre\_max\_1y\_site} (\emph{max.pre} in the manuscript):
  maximum of the monthly precipitation in the test sites during the year
  preceding the measurement (mm)
\item
  \texttt{tmx\_mean\_1y\_site} (\emph{mean.tmax} in the manuscript):
  mean of monthly maximum temperatures in the test sites during the year
  preceding the measurement (°C)
\item
  \texttt{bio1\_prov} (\emph{mean.temp} in the manuscript): average of
  the annual daily mean temperature in the provenances over the period
  from 1901 to 2009 (°C).
\item
  \texttt{bio5\_prov} (\emph{max.temp} in the manuscript): average of
  the maximum temperature of the warmest month in the provenances over
  the period from 1901 to 2009 (°C).
\item
  \texttt{bio12\_prov} (\emph{min.pre} in the manuscript): average of
  the precipitation of the driest month in the provenances over the
  period from 1901 to 2009 (mm).
\item
  \texttt{bio14\_prov} (\emph{mean.pre} in the manuscript): average of
  the annual precipitation in the provenances over the period from 1901
  to 2009 (mm).
\item
  \texttt{Q1}: proportion of assignment to the northern African (NA)
  gene pool for each clone.
\item
  \texttt{Q2}: proportion of assignment to the Corsican (C) gene pool
  for each clone.
\item
  \texttt{Q3}: proportion of assignment to the central Spain (CS) gene
  pool for each clone.
\item
  \texttt{Q4}: proportion of assignment to the French Atlantic (FA) gene
  pool for each clone.
\item
  \texttt{Q5}: proportion of assignment to the Iberian Atlantic (IA)
  gene pool for each clone.
\item
  \texttt{Q6}: proportion of assignment to the south-eastern Spain (SES)
  gene pool for each clone.
\item
  \texttt{max.Qvalue}: proportion of assignment to the main gene pool
  for each clone.
\item
  \texttt{max.Q}: main gene pool for each clone.
\item
  \texttt{P1}: qualitative variable indicating the assignment of each
  observation to the test or train dataset of the P1 partition.
\item
  \texttt{P2}: qualitative variable indicating the assignment of each
  observation to the test or train dataset of the P2 partition.
\item
  \texttt{P3}: qualitative variable indicating the assignment of each
  observation to the test or train dataset of the P3 partition.
\end{itemize}

\subsubsection{Genomic data}\label{genomic-data}

\textbf{In the dataset \texttt{GenomicData\_5165SNPs\_523clones.csv}.}

This file contains the genotype (noted as 0, 1 or 2) of each clone.
There are 5,165 SNPs in rows and 523 clones in columns. Missing data are
indicated with \emph{NA}.

\subsubsection{\texorpdfstring{piMASS outputs from de
\href{https://onlinelibrary.wiley.com/doi/full/10.1111/mec.16367?casa_token=1nNTc88Iy40AAAAA\%3ALd4EOK5ehk_cEHIkw5A9l8nk0NPzUzlYPX8eAjVCikIjHP0WJ1kxoHJSZjMLFsZcP-8wdbNuNrlOfp1jzw}{Miguel
et al.
(2022)}}{piMASS outputs from de Miguel et al. (2022)}}\label{pimass-outputs-from-de-miguel-et-al.-2022}

\textbf{In the dataset \texttt{height\_all\_sites\_res.mcmc.txt}.}

This file corresponds to the piMASS outputs of the Bayesian variable
selection regression {[}VSR{]} implemented in the piMASS software
(\href{https://projecteuclid.org/journals/annals-of-applied-statistics/volume-5/issue-3/Bayesian-variable-selection-regression-for-genome-wide-association-studies-and/10.1214/11-AOAS455.full}{Guan
\& Stephens 2011}) and which allows the identification of SNPs
associated with the phenotype (here BLUPs for height estimated across
the five CLONAPIN common gardens).

Accroding the
\href{https://www.haplotype.org/download/pimass-manual.pdf}{piMASS
manual}, the output file contains:

\begin{itemize}
\tightlist
\item
  \texttt{rs}: SNP ID.
\item
  \texttt{chr}: chromosome (no information on it in Miguel et al. 2022,
  so there are only `0').
\item
  \texttt{pos}: position.
\item
  \texttt{postc}: estimates of the posterior inclusion probabilities
  based on simple counting.
\item
  \texttt{postrb}: estimates of the posterior inclusion probabilities
  based on Rao-Blackwellization.
\item
  \texttt{beta}: the naive estimates of the posterior effect size.
\item
  \texttt{betarb}: \textbf{Rao-Blackwellized estimates of the posterior
  effect size, that are used in the present study to differentiate
  height-associated SNPs from SNPs not associated with height (i.e.
  `neutral' SNPs).}
\end{itemize}

\subsection{Intermediate files}\label{intermediate-files}

\subsubsection{Gene pool-specific GRMs}\label{gene-pool-specific-grms}

\textbf{Files \texttt{GRM\_AX.csv}, with \emph{X} being the gene pool
number.}

The gene pool-specific genomic relationship matrices (GRM) are
calculated in the script \texttt{2\_CalculateGenePoolSpecificGRM.R} and
are then used when fitting \emph{model M5}.

\subsection{Code}\label{code}

It was run on \emph{R version 3.6.3} and \emph{RStudio version 1.1.463}.

The code included in this repository constitutes the code necessary to
replicate the analyses in the paper published in \emph{The American
Naturalist} from the data in this same repository. An extended version
of the code can be found in the github repository
\url{https://github.com/JulietteArchambeau/HeightPinpinClonapin}, where
data sorting from a larger phenotypic database (inlcuding all traits
measured in the common gardens of the
\href{https://www6.bordeaux-aquitaine.inrae.fr/biogeco/Ressources/In-situ-dispositifs-de-terrain-observation-experimentation/Tests-de-provenances/CLONAPIN}{CLONAPIN
network}), climate and soil data extraction,additional exploratory
analyses, model output analyses and visualizations are included.

\end{document}
