---
title: "Model Posterior Distributions - P1 partition"
author: "Juliette Archambeau"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  html_document:
    toc: true
    toc_depth: 4
    toc_float:
       collapsed: false
    number_sections: true
    highlight: textmate
---

<style>
pre {
  overflow-x: auto;
}
pre code {
  word-wrap: normal;
  white-space: pre;
}
</style>

<style type="text/css">
div.main-container {
  max-width: 2000px;
  margin-left: auto;
  margin-right: auto;
}
</style>


```{css, echo=FALSE}
pre {
  max-height: 300px;
  overflow-y: auto;
}

pre[class] {
  max-height: 600px;
}
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width = 5,fig.height = 4,cache=F,echo=F)
options(width = 300)
library(knitr)
library(broom)
library(latex2exp)
library(cowplot)
library(ggplot2)
library(ggpubr)
library(stringi)
library(tidybayes)
library(dplyr)
library(bayesplot)
color_scheme_set("green")
library(devtools)
library(xtable)
library(ggridges)
library(tidyverse)
library(tibble)
library(brms)
```


<!-- Functions used -->

```{r funtions_used}
source("../../scripts/Functions/vir_lite.R")
square <- function(x) (x*x)

# Parameters of the funtion vir_lite
ds=0.7
```


In the Supplementary Information, we will report the medians of the posterior distributions, but both the medians and the means are shown in this document. 


The credible intervals used are 95% CI.

```{r CredibleIntervalsUsed}
prob=0.95
probs <- c((1 - prob) / 2, 1 - (1 - prob) / 2)
```


In the function 'mcmc_intervals':

```{r PosteriorIntervelsInFigures}
prob_outer = 0.99
```


```{r LoadDATA}
data <- readRDS(file="../../data/TrainP1.RDS")
```


# **Model M0**

```{r loadM0}
mod <- readRDS(file="../../outputs/models/P1/MOD0.rds")
```


## Main parameters

### Standard deviations

#### Mean

```{r TableM0SDMean, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```

#### Median

```{r TableM0SDMedian, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs

```{r M0MCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            "sd_site:block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_prov:clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            "sd_clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            "sd_prov__Intercept"=parse(text = TeX("$\\sigma_{P}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```



### Variances

#### Mean 

```{r TableM0VARMean, echo=F, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```

#### Median

In the Supplementary Information.

```{r TableM0VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M0_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```


#### Figs

In the Supplementary Information.

```{r M0VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_prov|sd_site:block|sigma)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov__Intercept)"=parse(text = TeX("$\\sigma^{2}_{P}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "sd_site__Intercept",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(1,0.3,0.7),nrow=3,vjust=c(1.3,0.8,1))
#ggsave(p,file="../../figs/SuppInfo/M0_PostMainVar.png",height=8) 
p
```


## Site intercepts

### Tables

#### Mean

```{r TableM0MeanSiteIntercepts}
df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_site\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term,'r_site[asturias,Intercept]'="$S_{Asturias}$",
                              'r_site[bordeaux,Intercept]'= '$S_{Bordeaux}$',
                              'r_site[caceres,Intercept]'='$S_{Caceres}$',
                              'r_site[madrid,Intercept]'='$S_{Madrid}$',
                              'r_site[portugal,Intercept]'="$S_{Portugal}$"))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

#### Median 

In the Supplementary Information.

```{r TableM0MedianSiteIntercepts}
df <- mod %>% broom::tidyMCMC(estimate.method = "median",conf.int = T,conf.level = 0.95) %>% 
  filter(str_detect(term, "^(r_site\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Median"=Estimate,"SD"=Std.error,"InfCI"=Conf.low,"SupCI"=Conf.high) %>% 
  mutate(Parameter = recode_factor(Term,'r_site[asturias,Intercept]'="$S_{Asturias}$",
                              'r_site[bordeaux,Intercept]'= '$S_{Bordeaux}$',
                              'r_site[caceres,Intercept]'='$S_{Caceres}$',
                              'r_site[madrid,Intercept]'='$S_{Madrid}$',
                              'r_site[portugal,Intercept]'="$S_{Portugal}$")) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)
df %>% knitr::kable(digits = 3 ,format = "markdown") 

# print(xtable(df, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M0_SiteInterceptsPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```


### Figs

#### MCMC intervals

```{r M0SiteInterceptsMCMCintervals, fig.width=8,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^r_site\\[",
                    prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() + 
  scale_y_discrete(labels=c('r_site[asturias,Intercept]'=parse(text = TeX("$S_{Asturias}$")),
                              'r_site[bordeaux,Intercept]'=parse(text = TeX("$S_{Bordeaux}$")),
                              'r_site[caceres,Intercept]'=parse(text = TeX("$S_{Caceres}$")),
                              'r_site[madrid,Intercept]'=parse(text = TeX("$S_{Madrid}$")),
                              'r_site[portugal,Intercept]'=parse(text = TeX("$S_{Portugal}$"))
                            )) +
  theme(axis.text = element_text(size=16))
```

#### Density ridges

In the Supplementary Information.

```{r M0SiteInterceptsDensityRidges, fig.height=5,fig.width=8,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_site\\[")
colnames(POST) <- str_sub(colnames(POST),8,-12) %>% str_to_title()
POST <- as.data.frame(t(POST))
POST$sites <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -sites)

sum <- posteriorsimpelmodellong  %>% 
   group_by(sites) %>%
   dplyr::summarize(mean=mean(value)) 

p <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = sites,
                                    fill   = as.factor(sites),
                                    vline_color = ..quantile..),
                                scale = 1.8, 
                                alpha = .8,
                                size=0.8,
                                rel_min_height=.001,
                                quantile_lines = TRUE, 
                                quantiles = c(0.025,0.5,0.975)) +
  scale_y_discrete(labels=c("Caceres"=parse(text = TeX("$S_{Caceres}$")),
                            "Bordeaux"=parse(text = TeX("$S_{Bordeaux}$")),
                            "Portugal"=parse(text = TeX("$S_{Portugal}$")),
                            "Madrid"=parse(text = TeX("$S_{Madrid}$")),
                            "Asturias"=parse(text = TeX("$S_{Asturias}$"))
                            )) +
  # coord_cartesian(c(-,0.8))+
  scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("5% & 95% quantiles", "mean"),
                        name = NULL) +
  labs(title="",
       y        = "", 
       x        = TeX("Estimate of the site intercept $S_{s}$")
       ) + 
  scale_fill_manual(values=c(vir_lite("cyan2",ds=ds),
                             vir_lite("navyblue",ds=ds),
                             vir_lite("pink",ds=ds),
                             vir_lite("deeppink",ds=ds),
                             vir_lite("dodgerblue2",ds=ds)
                             )) +
  theme_bw() + theme(axis.text = element_text(size=22),axis.title = element_text(size=22), 
                     legend.position = "none",plot.title = element_text(size=22))

p
# ggsave(p,file="../../figs/SuppInfo/M0_SiteInterceptsPost.png",height=7,width=12)
```



# **Model M1**

```{r loadM1}
mod <- readRDS(file="../../outputs/models/P1/MOD1.rds")
```




## Main parameters

### Standard deviations

#### Mean

```{r TableM1SDMean, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```

```{r TableM1SDMeanBis, eval=F}
# Same table as the previous one, but I kept it to check that I obtain the same numbers between the two. 
df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(b|b|sd|sigma)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term,'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

#### Median

```{r TableM1SDMedian, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


```{r TableM1SDMedianBis,eval=F}
df <- mod %>% broom::tidyMCMC(estimate.method = "median",conf.int = T) %>% 
  filter(str_detect(term, "^(b|b|sd|sigma)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Median"=Estimate) %>% 
  mutate(Term = recode_factor(Term,'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```



#### Figs

```{r M1SDFigs, fig.width=10,warning=F,message=F,eval=F}
POST <- posterior_samples(mod,pars = "^(b_a|b_Ia|sd|sigma)")
POST <- as.data.frame(t(POST))
POST$variables <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -variables)%>%
  group_by(variables) %>%
  dplyr::mutate(meanpervariables = mean(value))%>%
  ungroup()

p <- ggplot()+
  geom_vline(xintercept = 0, 
             col        = "grey70") +
  scale_y_discrete(labels=c("b_age.sc"="beta_age",
                            "b_Iage.scE2"="beta_age2",
                            "sd_prov:clon__Intercept"="sigma_genotype",
                            "sd_clon__Intercept"="sigma_genotype",
                            "sd_prov__Intercept"="sigma_provenance",
                            "sd_site__Intercept"="sigma_site",
                            "sd_site:block__Intercept"="sigma_block")) + 
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = variables,
                                    vline_color = ..quantile..,
                                    fill   = as.factor(variables)),
                                scale = 3, 
                                alpha = .8,
                                rel_min_height=.0001,
                                size=0.5,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  coord_cartesian(c(-0.25,1))+
  labs(y        = "", 
       x        = "Posterior distributions") + 
  scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("5th & 95th percentiles", "Median"),
                        name = NULL) +
  scale_fill_manual(values=c(vir_lite("cornflowerblue"),
                             vir_lite("cornflowerblue"),
                             vir_lite("magenta2"),
                             vir_lite("magenta2"),
                             vir_lite("green4"),
                             vir_lite("green4"),
                             vir_lite("gray80"))) +
  theme_bw() + theme(axis.text = element_text(size=16),axis.title = element_text(size=16), 
                      #  legend.text = element_text(size=18),legend.title = element_text(size=20)
                     legend.position = "none")
p
```

```{r M1MCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            "sd_site:block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_prov:clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            "sd_clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            "sd_prov__Intercept"=parse(text = TeX("$\\sigma_{P}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```



### Variances

#### Mean 

```{r TableM1VARMeanWrong, message =F, warning=F,eval=F}
# This table is wrong, because option 1  is not equal to option 2 (see below).

# Option 1 (wrong)
c(2,3,4) %>% mean() %>% square()

# Option 2 (ok)
c(2,3,4) %>% square() %>% mean()

# You have to 1) square the samples from the posterior, 2) calculate their mean,sd,intervals.

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(b|sd|sigma)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) 

bind_rows(df %>% dplyr::filter(str_detect(Term, "^(sd|sigma)")) %>% 
  mutate_if(is.numeric,square) %>% 
  mutate(Term = recode_factor(Term,'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'= '$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sigma'='$\\sigma^{2}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')),


df %>%  dplyr::filter(!str_detect(Term, "^(sd|sigma)")) %>%  
  mutate(Term = recode_factor(Term,'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$'))) %>% 
  
  knitr::kable(digits = 3 ,format = "markdown")
```

```{r TableM1VARMean, echo=F, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM1VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M1_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```




#### Figs

In the Supplementary Information.

```{r M1VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_prov|sd_site:block|sigma)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov__Intercept)"=parse(text = TeX("$\\sigma^{2}_{P}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "sd_site__Intercept",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(1,0.3,0.7),nrow=3,vjust=c(1.3,0.8,1))
ggsave(p,file="../../figs/SuppInfo/M1_PostMainVar.png",height=8) 
p
```




## Site intercepts

### Tables

#### Mean

```{r TableM1MeanSiteIntercepts}
df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_site\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term,'r_site[asturias,Intercept]'="$S_{Asturias}$",
                              'r_site[bordeaux,Intercept]'= '$S_{Bordeaux}$',
                              'r_site[caceres,Intercept]'='$S_{Caceres}$',
                              'r_site[madrid,Intercept]'='$S_{Madrid}$',
                              'r_site[portugal,Intercept]'="$S_{Portugal}$"))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

#### Median 

In the Supplementary Information.

```{r TableM1MedianSiteIntercepts}
df <- mod %>% broom::tidyMCMC(estimate.method = "median",conf.int = T,conf.level = 0.95) %>% 
  filter(str_detect(term, "^(r_site\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Median"=Estimate,"SD"=Std.error,"InfCI"=Conf.low,"SupCI"=Conf.high) %>% 
  mutate(Parameter = recode_factor(Term,'r_site[asturias,Intercept]'="$S_{Asturias}$",
                              'r_site[bordeaux,Intercept]'= '$S_{Bordeaux}$',
                              'r_site[caceres,Intercept]'='$S_{Caceres}$',
                              'r_site[madrid,Intercept]'='$S_{Madrid}$',
                              'r_site[portugal,Intercept]'="$S_{Portugal}$")) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)
df %>% knitr::kable(digits = 3 ,format = "markdown") 

print(xtable(df, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M1_SiteInterceptsPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```


### Figs

#### MCMC intervals

```{r M1SiteInterceptsMCMCintervals, fig.width=8,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^r_site\\[",
                    prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() + 
  scale_y_discrete(labels=c('r_site[asturias,Intercept]'=parse(text = TeX("$S_{Asturias}$")),
                              'r_site[bordeaux,Intercept]'=parse(text = TeX("$S_{Bordeaux}$")),
                              'r_site[caceres,Intercept]'=parse(text = TeX("$S_{Caceres}$")),
                              'r_site[madrid,Intercept]'=parse(text = TeX("$S_{Madrid}$")),
                              'r_site[portugal,Intercept]'=parse(text = TeX("$S_{Portugal}$"))
                            )) +
  theme(axis.text = element_text(size=16))
```

#### Density ridges

In the Supplementary Information.

```{r M1SiteInterceptsDensityRidges, fig.height=5,fig.width=8,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_site\\[")
colnames(POST) <- str_sub(colnames(POST),8,-12) %>% str_to_title()
POST <- as.data.frame(t(POST))
POST$sites <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -sites)

sum <- posteriorsimpelmodellong  %>% 
   group_by(sites) %>%
   dplyr::summarize(mean=mean(value)) 

p <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = sites,
                                    fill   = as.factor(sites),
                                    vline_color = ..quantile..),
                                scale = 1.8, 
                                alpha = .8,
                                size=0.8,
                                rel_min_height=.001,
                                quantile_lines = TRUE, 
                                quantiles = c(0.025,0.5,0.975)) +
  scale_y_discrete(labels=c("Caceres"=parse(text = TeX("$S_{Caceres}$")),
                            "Bordeaux"=parse(text = TeX("$S_{Bordeaux}$")),
                            "Portugal"=parse(text = TeX("$S_{Portugal}$")),
                            "Madrid"=parse(text = TeX("$S_{Madrid}$")),
                            "Asturias"=parse(text = TeX("$S_{Asturias}$"))
                            )) +
  # coord_cartesian(c(-,0.8))+
  scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("5% & 95% quantiles", "mean"),
                        name = NULL) +
  labs(title="",
       y        = "", 
       x        = TeX("Estimate of the site intercept $S_{s}$")
       ) + 
  scale_fill_manual(values=c(vir_lite("cyan2",ds=ds),
                             vir_lite("navyblue",ds=ds),
                             vir_lite("pink",ds=ds),
                             vir_lite("deeppink",ds=ds),
                             vir_lite("dodgerblue2",ds=ds)
                             )) +
  theme_bw() + theme(axis.text = element_text(size=22),axis.title = element_text(size=22), 
                     legend.position = "none",plot.title = element_text(size=22))

p
ggsave(p,file="../../figs/SuppInfo/M1_SiteInterceptsPost.png",height=7,width=12)
```



```{r M1SiteInterceptsDensityRidgesSVG, fig.height=5,fig.width=8,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_site\\[")
colnames(POST) <- str_sub(colnames(POST),8,-12) %>% str_to_title()
POST <- as.data.frame(t(POST))
POST$sites <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -sites)

sum <- posteriorsimpelmodellong  %>% 
   group_by(sites) %>%
   dplyr::summarize(mean=mean(value)) 

p <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = sites,
                                    fill   = as.factor(sites),
                                    #color=as.factor(sites),
                                    vline_color = ..quantile..),
                                scale = 1.8, 
                                alpha = .8,
                                size=0.8,
                                rel_min_height=.001,
                                quantile_lines = TRUE, 
                                quantiles = c(0.025,0.5,0.975)) +
  scale_y_discrete(labels=c("Caceres"=parse(text = TeX("$Caceres$")),
                            "Bordeaux"=parse(text = TeX("$Bordeaux$")),
                            "Portugal"=parse(text = TeX("$Portugal$")),
                            "Madrid"=parse(text = TeX("$Madrid$")),
                            "Asturias"=parse(text = TeX("$Asturias$"))
                            )) +
   coord_cartesian(ylim=c(1,6.4))+
  scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("5% & 95% quantiles", "mean"),
                        name = NULL) +
  labs(title="",
       y        = "", 
       x        = TeX("Estimate of the site intercept $S_{s}$")
       ) + 
  scale_fill_manual(values=c("gray40","gray40","gray40","gray40","gray40")) +
   # scale_color_manual(values=c("red","red","red","red","red")) +
  theme_bw() + theme(axis.text = element_text(size=22),axis.title = element_text(size=22), 
                     legend.position = "none",plot.title = element_text(size=22))

p
ggsave(p,file="../../figs/SuppInfo/M1_SiteInterceptsPost.svg",height=7,width=12)
```



## Conditional effect of age

In the Supplementary Information.

```{r M1AgeEffect,fig.width=5}
p <- plot(conditional_effects(mod,"age.sc"),plot=FALSE)[[1]] + 
  xlab("Mean-centered age") + 
  ylab("Logarithm of height (mm)") +
  theme_bw()
p
ggsave(p,file="../../figs/SuppInfo/M1_CondEffectAge.png",height=6,width=6)
```


## Provenance intercepts

### Table 

#### Mean 

```{r TableM1MeanProvIntercepts}
# provs <- unique(data$prov)
# par <-paste(paste0("'r_prov[",provs,",Intercept]'")) 
# par2 <-paste(paste0("'$P_{",provs,"}$'")) 
# tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_prov\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_prov[MIM,Intercept]' = '$P_{MIM}$','r_prov[CEN,Intercept]' = '$P_{CEN}$','r_prov[ORI,Intercept]' = '$P_{ORI}$','r_prov[STJ,Intercept]' = '$P_{STJ}$','r_prov[HOU,Intercept]' = '$P_{HOU}$','r_prov[CUE,Intercept]' = '$P_{CUE}$','r_prov[TAM,Intercept]' = '$P_{TAM}$','r_prov[LEI,Intercept]' = '$P_{LEI}$','r_prov[VER,Intercept]' = '$P_{VER}$','r_prov[CAS,Intercept]' = '$P_{CAS}$','r_prov[PIA,Intercept]' = '$P_{PIA}$','r_prov[ARM,Intercept]' = '$P_{ARM}$','r_prov[PET,Intercept]' = '$P_{PET}$','r_prov[VAL,Intercept]' = '$P_{VAL}$','r_prov[SAL,Intercept]' = '$P_{SAL}$','r_prov[OLO,Intercept]' = '$P_{OLO}$','r_prov[CAD,Intercept]' = '$P_{CAD}$','r_prov[ARN,Intercept]' = '$P_{ARN}$','r_prov[BAY,Intercept]' = '$P_{BAY}$','r_prov[SIE,Intercept]' = '$P_{SIE}$','r_prov[SEG,Intercept]' = '$P_{SEG}$','r_prov[PLE,Intercept]' = '$P_{PLE}$','r_prov[BON,Intercept]' = '$P_{BON}$','r_prov[COC,Intercept]' = '$P_{COC}$','r_prov[SAC,Intercept]' = '$P_{SAC}$','r_prov[QUA,Intercept]' = '$P_{QUA}$','r_prov[CAR,Intercept]' = '$P_{CAR}$','r_prov[OLB,Intercept]' = '$P_{OLB}$','r_prov[PIE,Intercept]' = '$P_{PIE}$','r_prov[PUE,Intercept]' = '$P_{PUE}$','r_prov[ALT,Intercept]' = '$P_{ALT}$','r_prov[LAM,Intercept]' = '$P_{LAM}$','r_prov[MAD,Intercept]' = '$P_{MAD}$','r_prov[COM,Intercept]' = '$P_{COM}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```


#### Median

In the Supplementary Information.

```{r TableM1MedianProvIntercepts}
df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_prov\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Median"=Estimate,"SD"=Std.error,"InfCI"=Conf.low,"SupCI"=Conf.high) %>% 
  mutate(Parameter= recode_factor(Term, 'r_prov[MIM,Intercept]' = '$P_{MIM}$','r_prov[CEN,Intercept]' = '$P_{CEN}$','r_prov[ORI,Intercept]' = '$P_{ORI}$','r_prov[STJ,Intercept]' = '$P_{STJ}$','r_prov[HOU,Intercept]' = '$P_{HOU}$','r_prov[CUE,Intercept]' = '$P_{CUE}$','r_prov[TAM,Intercept]' = '$P_{TAM}$','r_prov[LEI,Intercept]' = '$P_{LEI}$','r_prov[VER,Intercept]' = '$P_{VER}$','r_prov[CAS,Intercept]' = '$P_{CAS}$','r_prov[PIA,Intercept]' = '$P_{PIA}$','r_prov[ARM,Intercept]' = '$P_{ARM}$','r_prov[PET,Intercept]' = '$P_{PET}$','r_prov[VAL,Intercept]' = '$P_{VAL}$','r_prov[SAL,Intercept]' = '$P_{SAL}$','r_prov[OLO,Intercept]' = '$P_{OLO}$','r_prov[CAD,Intercept]' = '$P_{CAD}$','r_prov[ARN,Intercept]' = '$P_{ARN}$','r_prov[BAY,Intercept]' = '$P_{BAY}$','r_prov[SIE,Intercept]' = '$P_{SIE}$','r_prov[SEG,Intercept]' = '$P_{SEG}$','r_prov[PLE,Intercept]' = '$P_{PLE}$','r_prov[BON,Intercept]' = '$P_{BON}$','r_prov[COC,Intercept]' = '$P_{COC}$','r_prov[SAC,Intercept]' = '$P_{SAC}$','r_prov[QUA,Intercept]' = '$P_{QUA}$','r_prov[CAR,Intercept]' = '$P_{CAR}$','r_prov[OLB,Intercept]' = '$P_{OLB}$','r_prov[PIE,Intercept]' = '$P_{PIE}$','r_prov[PUE,Intercept]' = '$P_{PUE}$','r_prov[ALT,Intercept]' = '$P_{ALT}$','r_prov[LAM,Intercept]' = '$P_{LAM}$','r_prov[MAD,Intercept]' = '$P_{MAD}$','r_prov[COM,Intercept]' = '$P_{COM}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)

df %>% knitr::kable(digits = 3 ,format = "markdown")

print(xtable(df, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M1_ProvInterceptsPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```


### Figs

<!-- #### MCMC intervals -->

```{r M1ProvInterceptsMCMCintervals, fig.height=8,fig.width=10,warning=F,message=F,eval=F}
mod %>%  mcmc_intervals(regex_pars = "^r_prov\\[",
                    prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() + 

  theme(axis.text = element_text(size=16))
```

#### Density Ridges

```{r M1ProvInterceptsDensityRidges, fig.width=10,fig.height=8,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov\\[")
colnames(POST) <- str_sub(colnames(POST),8,-12)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- droplevels(data)
ps <- data %>%
  dplyr::group_by(prov) %>% 
  dplyr::summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

p <- ggplot()+
  geom_vline(xintercept = 0, 
             col        = "grey70") +
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov)),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=.01,
                                size=0.3,quantile_lines = TRUE, quantiles = c(0.05,0.95)) +
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$P_{ALT}$")),
                            "ARM"=parse(text = TeX("$P_{ARM}$")),
                            "ARN"=parse(text = TeX("$P_{ARN}$")),
                            "BAY"=parse(text = TeX("$P_{BAY}$")),
                            "BON"=parse(text = TeX("$P_{BON}$")),
                            "CAD"=parse(text = TeX("$P_{CAD}$")),
                            "CAR"=parse(text = TeX("$P_{CAR}$")),
                            "CAS"=parse(text = TeX("$P_{CAS}$")),
                            "CEN"=parse(text = TeX("$P_{CEN}$")),
                            "COC"=parse(text = TeX("$P_{COC}$")),
                            "COM"=parse(text = TeX("$P_{COM}$")),
                            "CUE"=parse(text = TeX("$P_{CUE}$")),
                            "HOU"=parse(text = TeX("$P_{HOU}$")),
                            "LAM"=parse(text = TeX("$P_{LAM}$")),
                            "LEI"=parse(text = TeX("$P_{LEI}$")),
                            "MAD"=parse(text = TeX("$P_{MAD}$")),
                            "MIM"=parse(text = TeX("$P_{MIM}$")),
                            "OLB"=parse(text = TeX("$P_{OLB}$")),
                            "OLO"=parse(text = TeX("$P_{OLO}$")),
                            "ORI"=parse(text = TeX("$P_{ORI}$")),
                            "PET"=parse(text = TeX("$P_{PET}$")),
                            "PIA"=parse(text = TeX("$P_{PIA}$")),
                            "PIE"=parse(text = TeX("$P_{PIE}$")),
                            "PLE"=parse(text = TeX("$P_{PLE}$")),
                            "PUE"=parse(text = TeX("$P_{PUE}$")),
                            "QUA"=parse(text = TeX("$P_{QUA}$")),
                            "SAC"=parse(text = TeX("$P_{SAC}$")),
                            "SAL"=parse(text = TeX("$P_{SAL}$")),
                            "SEG"=parse(text = TeX("$P_{SEG}$")),
                            "SIE"=parse(text = TeX("$P_{SIE}$")),
                            "STJ"=parse(text = TeX("$P_{STJ}$")),
                            "TAM"=parse(text = TeX("$P_{TAM}$")),
                            "VAL"=parse(text = TeX("$P_{VAL}$")),
                            "VER"=parse(text = TeX("$P_{VER}$")))) +
   coord_cartesian(xlim=c(-0.4,0.25),ylim=c(0.5,36.5))+ # ,ylim=c(1,37)
  scale_fill_manual(values=c("orangered3",
                             "gold2",
                             "darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Northern Africa", 
                                                   "Corsica",
                                                   "Central Spain",
                                                   "French Atlantic",
                                                   "Iberian Atlantic",
                                                   "South-eastern Spain"),name="Gene pools") +
  labs(fill     = "Gene pools",
       y        = "", 
       x        = TeX("Estimate of the provenance intercept P_{p}")
       )+
  theme_bw() + theme(axis.text = element_text(size=15),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
p
ggsave(p,file="../../figs/SuppInfo/M1_ProvInterceptsPost.png",height=10,width=10)
```

```{r M1ProvInterceptsDensityRidgesbis, fig.width=10,fig.height=8,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov\\[")
colnames(POST) <- str_sub(colnames(POST),8,-12)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- droplevels(data)
ps <- data %>%
  dplyr::group_by(prov) %>% 
  dplyr::summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

p <- ggplot()+
  geom_vline(xintercept = 0, 
             col        = "grey70") +
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov)),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=.01,
                                size=0.3,quantile_lines = TRUE, quantiles = c(0.05,0.95)) +
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$ALT$")),
                            "ARM"=parse(text = TeX("$ARM$")),
                            "ARN"=parse(text = TeX("$ARN$")),
                            "BAY"=parse(text = TeX("$BAY$")),
                            "BON"=parse(text = TeX("$BON$")),
                            "CAD"=parse(text = TeX("$CAD$")),
                            "CAR"=parse(text = TeX("$CAR$")),
                            "CAS"=parse(text = TeX("$CAS$")),
                            "CEN"=parse(text = TeX("$CEN$")),
                            "COC"=parse(text = TeX("$COC$")),
                            "COM"=parse(text = TeX("$COM$")),
                            "CUE"=parse(text = TeX("$CUE$")),
                            "HOU"=parse(text = TeX("$HOU$")),
                            "LAM"=parse(text = TeX("$LAM$")),
                            "LEI"=parse(text = TeX("$LEI$")),
                            "MAD"=parse(text = TeX("$MAD$")),
                            "MIM"=parse(text = TeX("$MIM$")),
                            "OLB"=parse(text = TeX("$OLB$")),
                            "OLO"=parse(text = TeX("$OLO$")),
                            "ORI"=parse(text = TeX("$ORI$")),
                            "PET"=parse(text = TeX("$PET$")),
                            "PIA"=parse(text = TeX("$PIA$")),
                            "PIE"=parse(text = TeX("$PIE$")),
                            "PLE"=parse(text = TeX("$PLE$")),
                            "PUE"=parse(text = TeX("$PUE$")),
                            "QUA"=parse(text = TeX("$QUA$")),
                            "SAC"=parse(text = TeX("$SAC$")),
                            "SAL"=parse(text = TeX("$SAL$")),
                            "SEG"=parse(text = TeX("$SEG$")),
                            "SIE"=parse(text = TeX("$SIE$")),
                            "STJ"=parse(text = TeX("$STJ$")),
                            "TAM"=parse(text = TeX("$TAM$")),
                            "VAL"=parse(text = TeX("$VAL$")),
                            "VER"=parse(text = TeX("$VER$")))) +
   coord_cartesian(xlim=c(-0.4,0.25),ylim=c(0.5,36.5))+ # ,ylim=c(1,37)
  scale_fill_manual(values=c("orangered3",
                             "gold2",
                             "darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Northern Africa", 
                                                   "Corsica",
                                                   "Central Spain",
                                                   "French Atlantic",
                                                   "Iberian Atlantic",
                                                   "South-eastern Spain"),name="Gene pools") +
  labs(fill     = "Gene pools",
       y        = "", 
       x        = TeX("Estimate of the provenance intercept P_{p}")
       )+
  theme_bw() + theme(axis.text = element_text(size=15),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
p
ggsave(p,file="../../figs/SuppInfo/M1_ProvInterceptsPost.svg",height=10,width=10)
```






# **Model M2**

```{r loadM2}
mod <- readRDS(file="../../outputs/models/P1/MOD2.rds")
```

## Main parameters

### Standard deviations

#### Mean

```{r TableM2SDMean, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov:site__Intercept'='$\\sigma_{Inter}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Median

```{r TableM2SDMedian, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov:site__Intercept'='$\\sigma_{Inter}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs

```{r M2MCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  
    scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            "sd_site:block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_prov:clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            "sd_clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            "sd_prov__Intercept"=parse(text = TeX("$\\sigma_{P}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            'sd_prov:site__Intercept'=parse(text = TeX("$\\sigma_{Inter}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM2VARMean, echo=F, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_prov:site__Intercept'='$\\sigma^{2}_{Inter}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM2VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_prov:site__Intercept'='$\\sigma^{2}_{Inter}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M2_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```


#### Figs

In the Supplementary Information.

```{r M2VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_prov|sd_site:block|sigma)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov__Intercept)"=parse(text = TeX("$\\sigma^{2}_{P}$")),
                            "square(sd_prov:site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{Inter}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "sd_site__Intercept",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(1,0.3,0.7),nrow=3,vjust=c(1.3,0.8,1))
ggsave(p,file="../../figs/SuppInfo/M2_PostMainVar.png",height=8) 
p
```



## Site intercepts

### Tables

#### Mean

```{r TableM2MeanSiteIntercepts}
df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T,conf.level = 0.95) %>% 
  filter(str_detect(term, "^(r_site\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate,"SD"=Std.error,"InfCI"=Conf.low,"SupCI"=Conf.high) %>% 
  mutate(Parameter = recode_factor(Term,'r_site[asturias,Intercept]'="$S_{Asturias}$",
                              'r_site[bordeaux,Intercept]'= '$S_{Bordeaux}$',
                              'r_site[caceres,Intercept]'='$S_{Caceres}$',
                              'r_site[madrid,Intercept]'='$S_{Madrid}$',
                              'r_site[portugal,Intercept]'="$S_{Portugal}$")) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

#### Median 

```{r TableM2MedianSiteIntercepts}
df <- mod %>% broom::tidyMCMC(estimate.method = "median",conf.int = T,conf.level = 0.95) %>% 
  filter(str_detect(term, "^(r_site\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Median"=Estimate,"SD"=Std.error,"InfCI"=Conf.low,"SupCI"=Conf.high) %>% 
  mutate(Parameter = recode_factor(Term,'r_site[asturias,Intercept]'="$S_{Asturias}$",
                              'r_site[bordeaux,Intercept]'= '$S_{Bordeaux}$',
                              'r_site[caceres,Intercept]'='$S_{Caceres}$',
                              'r_site[madrid,Intercept]'='$S_{Madrid}$',
                              'r_site[portugal,Intercept]'="$S_{Portugal}$")) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)
df %>% knitr::kable(digits = 3 ,format = "markdown")
```


### Figs

#### MCMC intervals

```{r M2SiteInterceptsMCMCintervals, fig.width=8,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^r_site\\[",
                    prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() + 
  scale_y_discrete(labels=c('r_site[asturias,Intercept]'=parse(text = TeX("$S_{Asturias}$")),
                              'r_site[bordeaux,Intercept]'=parse(text = TeX("$S_{Bordeaux}$")),
                              'r_site[caceres,Intercept]'=parse(text = TeX("$S_{Caceres}$")),
                              'r_site[madrid,Intercept]'=parse(text = TeX("$S_{Madrid}$")),
                              'r_site[portugal,Intercept]'=parse(text = TeX("$S_{Portugal}$"))
                            )) +
  theme(axis.text = element_text(size=16))
```



## Provenance intercepts

### ALL

```{r M2AllProvs, warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov\\[")
colnames(POST) <- str_sub(colnames(POST),8,-12)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- readRDS(file="../../../data/cleaned-data/datatrain_pheno-clim-soil_20012020_Sample2_75percent.RDS")
data <- droplevels(data)
ps <- data %>%
  group_by(prov) %>% 
  summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

pm2_all <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov),
                                    vline_color = ..quantile..),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=c(.01),
                                size=0.2,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$P_{ALT}$")),
                            "ARM"=parse(text = TeX("$P_{ARM}$")),
                            "ARN"=parse(text = TeX("$P_{ARN}$")),
                            "BAY"=parse(text = TeX("$P_{BAY}$")),
                            "BON"=parse(text = TeX("$P_{BON}$")),
                            "CAD"=parse(text = TeX("$P_{CAD}$")),
                            "CAR"=parse(text = TeX("$P_{CAR}$")),
                            "CAS"=parse(text = TeX("$P_{CAS}$")),
                            "CEN"=parse(text = TeX("$P_{CEN}$")),
                            "COC"=parse(text = TeX("$P_{COC}$")),
                            "COM"=parse(text = TeX("$P_{COM}$")),
                            "CUE"=parse(text = TeX("$P_{CUE}$")),
                            "HOU"=parse(text = TeX("$P_{HOU}$")),
                            "LAM"=parse(text = TeX("$P_{LAM}$")),
                            "LEI"=parse(text = TeX("$P_{LEI}$")),
                            "MAD"=parse(text = TeX("$P_{MAD}$")),
                            "MIM"=parse(text = TeX("$P_{MIM}$")),
                            "OLB"=parse(text = TeX("$P_{OLB}$")),
                            "OLO"=parse(text = TeX("$P_{OLO}$")),
                            "ORI"=parse(text = TeX("$P_{ORI}$")),
                            "PET"=parse(text = TeX("$P_{PET}$")),
                            "PIA"=parse(text = TeX("$P_{PIA}$")),
                            "PIE"=parse(text = TeX("$P_{PIE}$")),
                            "PLE"=parse(text = TeX("$P_{PLE}$")),
                            "PUE"=parse(text = TeX("$P_{PUE}$")),
                            "QUA"=parse(text = TeX("$P_{QUA}$")),
                            "SAC"=parse(text = TeX("$P_{SAC}$")),
                            "SAL"=parse(text = TeX("$P_{SAL}$")),
                            "SEG"=parse(text = TeX("$P_{SEG}$")),
                            "SIE"=parse(text = TeX("$P_{SIE}$")),
                            "STJ"=parse(text = TeX("$P_{STJ}$")),
                            "TAM"=parse(text = TeX("$P_{TAM}$")),
                            "VAL"=parse(text = TeX("$P_{VAL}$")),
                            "VER"=parse(text = TeX("$P_{VER}$")))) +
  
   coord_cartesian(c(-0.35,0.3))+
  
   scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Mean"),
                        name = NULL) +
  scale_fill_manual(values=c("orangered3",
                             "gold2","darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Q1: Northern Africa", 
                                                   "Q2: Corsica",
                                                   "Q3: Central Spain",
                                                   "Q4: French Atlantic",
                                                   "Q5: Iberian Atlantic",
                                                   "Q6: South-eastern Spain"),name="Gene pools") +
  labs(title     = "All sites",
       y        = "", 
       x        = TeX("Intercepts P_{p}")
       )+
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
```



### Portugal

```{r M2Portugal,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov:site\\[.*portugal")
colnames(POST) <- str_sub(colnames(POST),13,-21)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- readRDS(file="../../../data/cleaned-data/datatrain_pheno-clim-soil_20012020_Sample2_75percent.RDS")
data <- droplevels(data)
ps <- data %>%
  group_by(prov) %>% 
  summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

pm2_portugal <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov),
                                    vline_color = ..quantile..),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=c(.01),
                                size=0.2,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$P_{ALT}$")),
                            "ARM"=parse(text = TeX("$P_{ARM}$")),
                            "ARN"=parse(text = TeX("$P_{ARN}$")),
                            "BAY"=parse(text = TeX("$P_{BAY}$")),
                            "BON"=parse(text = TeX("$P_{BON}$")),
                            "CAD"=parse(text = TeX("$P_{CAD}$")),
                            "CAR"=parse(text = TeX("$P_{CAR}$")),
                            "CAS"=parse(text = TeX("$P_{CAS}$")),
                            "CEN"=parse(text = TeX("$P_{CEN}$")),
                            "COC"=parse(text = TeX("$P_{COC}$")),
                            "COM"=parse(text = TeX("$P_{COM}$")),
                            "CUE"=parse(text = TeX("$P_{CUE}$")),
                            "HOU"=parse(text = TeX("$P_{HOU}$")),
                            "LAM"=parse(text = TeX("$P_{LAM}$")),
                            "LEI"=parse(text = TeX("$P_{LEI}$")),
                            "MAD"=parse(text = TeX("$P_{MAD}$")),
                            "MIM"=parse(text = TeX("$P_{MIM}$")),
                            "OLB"=parse(text = TeX("$P_{OLB}$")),
                            "OLO"=parse(text = TeX("$P_{OLO}$")),
                            "ORI"=parse(text = TeX("$P_{ORI}$")),
                            "PET"=parse(text = TeX("$P_{PET}$")),
                            "PIA"=parse(text = TeX("$P_{PIA}$")),
                            "PIE"=parse(text = TeX("$P_{PIE}$")),
                            "PLE"=parse(text = TeX("$P_{PLE}$")),
                            "PUE"=parse(text = TeX("$P_{PUE}$")),
                            "QUA"=parse(text = TeX("$P_{QUA}$")),
                            "SAC"=parse(text = TeX("$P_{SAC}$")),
                            "SAL"=parse(text = TeX("$P_{SAL}$")),
                            "SEG"=parse(text = TeX("$P_{SEG}$")),
                            "SIE"=parse(text = TeX("$P_{SIE}$")),
                            "STJ"=parse(text = TeX("$P_{STJ}$")),
                            "TAM"=parse(text = TeX("$P_{TAM}$")),
                            "VAL"=parse(text = TeX("$P_{VAL}$")),
                            "VER"=parse(text = TeX("$P_{VER}$")))) +
  
   coord_cartesian(c(-0.35,0.3))+
  
   scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Mean"),
                        name = NULL) +
  scale_fill_manual(values=c("orangered3",
                             "gold2","darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Q1: Northern Africa", 
                                                   "Q2: Corsica",
                                                   "Q3: Central Spain",
                                                   "Q4: French Atlantic",
                                                   "Q5: Iberian Atlantic",
                                                   "Q6: South-eastern Spain"),name="Gene pools") +
  labs(title     = "Portugal",
       y        = "", 
       x        = TeX("Intercepts P_{p,Portugal}")
       )+
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
```

### Caceres


```{r M2Caceres, warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov:site\\[.*caceres")
colnames(POST) <- str_sub(colnames(POST),13,-20)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- readRDS(file="../../../data/cleaned-data/datatrain_pheno-clim-soil_20012020_Sample2_75percent.RDS")
data <- droplevels(data)
ps <- data %>%
  group_by(prov) %>% 
  summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

pm2_caceres <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov),
                                    vline_color = ..quantile..),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=c(.01),
                                size=0.2,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$P_{ALT}$")),
                            "ARM"=parse(text = TeX("$P_{ARM}$")),
                            "ARN"=parse(text = TeX("$P_{ARN}$")),
                            "BAY"=parse(text = TeX("$P_{BAY}$")),
                            "BON"=parse(text = TeX("$P_{BON}$")),
                            "CAD"=parse(text = TeX("$P_{CAD}$")),
                            "CAR"=parse(text = TeX("$P_{CAR}$")),
                            "CAS"=parse(text = TeX("$P_{CAS}$")),
                            "CEN"=parse(text = TeX("$P_{CEN}$")),
                            "COC"=parse(text = TeX("$P_{COC}$")),
                            "COM"=parse(text = TeX("$P_{COM}$")),
                            "CUE"=parse(text = TeX("$P_{CUE}$")),
                            "HOU"=parse(text = TeX("$P_{HOU}$")),
                            "LAM"=parse(text = TeX("$P_{LAM}$")),
                            "LEI"=parse(text = TeX("$P_{LEI}$")),
                            "MAD"=parse(text = TeX("$P_{MAD}$")),
                            "MIM"=parse(text = TeX("$P_{MIM}$")),
                            "OLB"=parse(text = TeX("$P_{OLB}$")),
                            "OLO"=parse(text = TeX("$P_{OLO}$")),
                            "ORI"=parse(text = TeX("$P_{ORI}$")),
                            "PET"=parse(text = TeX("$P_{PET}$")),
                            "PIA"=parse(text = TeX("$P_{PIA}$")),
                            "PIE"=parse(text = TeX("$P_{PIE}$")),
                            "PLE"=parse(text = TeX("$P_{PLE}$")),
                            "PUE"=parse(text = TeX("$P_{PUE}$")),
                            "QUA"=parse(text = TeX("$P_{QUA}$")),
                            "SAC"=parse(text = TeX("$P_{SAC}$")),
                            "SAL"=parse(text = TeX("$P_{SAL}$")),
                            "SEG"=parse(text = TeX("$P_{SEG}$")),
                            "SIE"=parse(text = TeX("$P_{SIE}$")),
                            "STJ"=parse(text = TeX("$P_{STJ}$")),
                            "TAM"=parse(text = TeX("$P_{TAM}$")),
                            "VAL"=parse(text = TeX("$P_{VAL}$")),
                            "VER"=parse(text = TeX("$P_{VER}$")))) +
   coord_cartesian(c(-0.35,0.3))+
  
   scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Mean"),
                        name = NULL) +
  scale_fill_manual(values=c("orangered3",
                             "gold2","darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Q1: Northern Africa", 
                                                   "Q2: Corsica",
                                                   "Q3: Central Spain",
                                                   "Q4: French Atlantic",
                                                   "Q5: Iberian Atlantic",
                                                   "Q6: South-eastern Spain"),name="Gene pools") +
  labs(title     = "Caceres",
       y        = "", 
       x        = TeX("Intercepts P_{p,Caceres}")
       )+
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
```



### Madrid


```{r M2Madrid, warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov:site\\[.*madrid")
colnames(POST) <- str_sub(colnames(POST),13,-19)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- readRDS(file="../../../data/cleaned-data/datatrain_pheno-clim-soil_20012020_Sample2_75percent.RDS")
data <- droplevels(data)
ps <- data %>%
  group_by(prov) %>% 
  summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

pm2_madrid <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov),
                                    vline_color = ..quantile..),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=c(.01),
                                size=0.2,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  
   coord_cartesian(c(-0.35,0.3))+
  
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$P_{ALT}$")),
                            "ARM"=parse(text = TeX("$P_{ARM}$")),
                            "ARN"=parse(text = TeX("$P_{ARN}$")),
                            "BAY"=parse(text = TeX("$P_{BAY}$")),
                            "BON"=parse(text = TeX("$P_{BON}$")),
                            "CAD"=parse(text = TeX("$P_{CAD}$")),
                            "CAR"=parse(text = TeX("$P_{CAR}$")),
                            "CAS"=parse(text = TeX("$P_{CAS}$")),
                            "CEN"=parse(text = TeX("$P_{CEN}$")),
                            "COC"=parse(text = TeX("$P_{COC}$")),
                            "COM"=parse(text = TeX("$P_{COM}$")),
                            "CUE"=parse(text = TeX("$P_{CUE}$")),
                            "HOU"=parse(text = TeX("$P_{HOU}$")),
                            "LAM"=parse(text = TeX("$P_{LAM}$")),
                            "LEI"=parse(text = TeX("$P_{LEI}$")),
                            "MAD"=parse(text = TeX("$P_{MAD}$")),
                            "MIM"=parse(text = TeX("$P_{MIM}$")),
                            "OLB"=parse(text = TeX("$P_{OLB}$")),
                            "OLO"=parse(text = TeX("$P_{OLO}$")),
                            "ORI"=parse(text = TeX("$P_{ORI}$")),
                            "PET"=parse(text = TeX("$P_{PET}$")),
                            "PIA"=parse(text = TeX("$P_{PIA}$")),
                            "PIE"=parse(text = TeX("$P_{PIE}$")),
                            "PLE"=parse(text = TeX("$P_{PLE}$")),
                            "PUE"=parse(text = TeX("$P_{PUE}$")),
                            "QUA"=parse(text = TeX("$P_{QUA}$")),
                            "SAC"=parse(text = TeX("$P_{SAC}$")),
                            "SAL"=parse(text = TeX("$P_{SAL}$")),
                            "SEG"=parse(text = TeX("$P_{SEG}$")),
                            "SIE"=parse(text = TeX("$P_{SIE}$")),
                            "STJ"=parse(text = TeX("$P_{STJ}$")),
                            "TAM"=parse(text = TeX("$P_{TAM}$")),
                            "VAL"=parse(text = TeX("$P_{VAL}$")),
                            "VER"=parse(text = TeX("$P_{VER}$")))) +
   scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Mean"),
                        name = NULL) +
  scale_fill_manual(values=c("orangered3",
                             "gold2","darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Q1: Northern Africa", 
                                                   "Q2: Corsica",
                                                   "Q3: Central Spain",
                                                   "Q4: French Atlantic",
                                                   "Q5: Iberian Atlantic",
                                                   "Q6: South-eastern Spain"),name="Gene pools") +
  labs(title     = "Madrid",
       y        = "", 
       x        = TeX("Intercepts P_{p,Madrid}")
       )+
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
```


### Bordeaux


```{r M2Bordeaux, warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov:site\\[.*bordeaux")
colnames(POST) <- str_sub(colnames(POST),13,-21)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- readRDS(file="../../../data/cleaned-data/datatrain_pheno-clim-soil_20012020_Sample2_75percent.RDS")
data <- droplevels(data)
ps <- data %>%
  group_by(prov) %>% 
  summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

pm2_bordeaux <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov),
                                    vline_color = ..quantile..),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=c(.01),
                                size=0.2,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  
   coord_cartesian(c(-0.35,0.3))+
  
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$P_{ALT}$")),
                            "ARM"=parse(text = TeX("$P_{ARM}$")),
                            "ARN"=parse(text = TeX("$P_{ARN}$")),
                            "BAY"=parse(text = TeX("$P_{BAY}$")),
                            "BON"=parse(text = TeX("$P_{BON}$")),
                            "CAD"=parse(text = TeX("$P_{CAD}$")),
                            "CAR"=parse(text = TeX("$P_{CAR}$")),
                            "CAS"=parse(text = TeX("$P_{CAS}$")),
                            "CEN"=parse(text = TeX("$P_{CEN}$")),
                            "COC"=parse(text = TeX("$P_{COC}$")),
                            "COM"=parse(text = TeX("$P_{COM}$")),
                            "CUE"=parse(text = TeX("$P_{CUE}$")),
                            "HOU"=parse(text = TeX("$P_{HOU}$")),
                            "LAM"=parse(text = TeX("$P_{LAM}$")),
                            "LEI"=parse(text = TeX("$P_{LEI}$")),
                            "MAD"=parse(text = TeX("$P_{MAD}$")),
                            "MIM"=parse(text = TeX("$P_{MIM}$")),
                            "OLB"=parse(text = TeX("$P_{OLB}$")),
                            "OLO"=parse(text = TeX("$P_{OLO}$")),
                            "ORI"=parse(text = TeX("$P_{ORI}$")),
                            "PET"=parse(text = TeX("$P_{PET}$")),
                            "PIA"=parse(text = TeX("$P_{PIA}$")),
                            "PIE"=parse(text = TeX("$P_{PIE}$")),
                            "PLE"=parse(text = TeX("$P_{PLE}$")),
                            "PUE"=parse(text = TeX("$P_{PUE}$")),
                            "QUA"=parse(text = TeX("$P_{QUA}$")),
                            "SAC"=parse(text = TeX("$P_{SAC}$")),
                            "SAL"=parse(text = TeX("$P_{SAL}$")),
                            "SEG"=parse(text = TeX("$P_{SEG}$")),
                            "SIE"=parse(text = TeX("$P_{SIE}$")),
                            "STJ"=parse(text = TeX("$P_{STJ}$")),
                            "TAM"=parse(text = TeX("$P_{TAM}$")),
                            "VAL"=parse(text = TeX("$P_{VAL}$")),
                            "VER"=parse(text = TeX("$P_{VER}$")))) +
   scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Mean"),
                        name = NULL) +
  scale_fill_manual(values=c("orangered3",
                             "gold2","darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Q1: Northern Africa", 
                                                   "Q2: Corsica",
                                                   "Q3: Central Spain",
                                                   "Q4: French Atlantic",
                                                   "Q5: Iberian Atlantic",
                                                   "Q6: South-eastern Spain"),name="Gene pools") +
  labs(title = "Bordeaux",
       y        = "", 
       x        = TeX("Intercepts P_{p,Bordeaux}")
       )+
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
```


### Asturias


```{r M2Asturias, warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov:site\\[.*asturias")
colnames(POST) <- str_sub(colnames(POST),13,-21)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- readRDS(file="../../../data/cleaned-data/datatrain_pheno-clim-soil_20012020_Sample2_75percent.RDS")
data <- droplevels(data)
ps <- data %>%
  group_by(prov) %>% 
  summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

pm2_asturias <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov),
                                    vline_color = ..quantile..),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=c(.01),
                                size=0.2,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  
   coord_cartesian(c(-0.35,0.3))+
  
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$P_{ALT}$")),
                            "ARM"=parse(text = TeX("$P_{ARM}$")),
                            "ARN"=parse(text = TeX("$P_{ARN}$")),
                            "BAY"=parse(text = TeX("$P_{BAY}$")),
                            "BON"=parse(text = TeX("$P_{BON}$")),
                            "CAD"=parse(text = TeX("$P_{CAD}$")),
                            "CAR"=parse(text = TeX("$P_{CAR}$")),
                            "CAS"=parse(text = TeX("$P_{CAS}$")),
                            "CEN"=parse(text = TeX("$P_{CEN}$")),
                            "COC"=parse(text = TeX("$P_{COC}$")),
                            "COM"=parse(text = TeX("$P_{COM}$")),
                            "CUE"=parse(text = TeX("$P_{CUE}$")),
                            "HOU"=parse(text = TeX("$P_{HOU}$")),
                            "LAM"=parse(text = TeX("$P_{LAM}$")),
                            "LEI"=parse(text = TeX("$P_{LEI}$")),
                            "MAD"=parse(text = TeX("$P_{MAD}$")),
                            "MIM"=parse(text = TeX("$P_{MIM}$")),
                            "OLB"=parse(text = TeX("$P_{OLB}$")),
                            "OLO"=parse(text = TeX("$P_{OLO}$")),
                            "ORI"=parse(text = TeX("$P_{ORI}$")),
                            "PET"=parse(text = TeX("$P_{PET}$")),
                            "PIA"=parse(text = TeX("$P_{PIA}$")),
                            "PIE"=parse(text = TeX("$P_{PIE}$")),
                            "PLE"=parse(text = TeX("$P_{PLE}$")),
                            "PUE"=parse(text = TeX("$P_{PUE}$")),
                            "QUA"=parse(text = TeX("$P_{QUA}$")),
                            "SAC"=parse(text = TeX("$P_{SAC}$")),
                            "SAL"=parse(text = TeX("$P_{SAL}$")),
                            "SEG"=parse(text = TeX("$P_{SEG}$")),
                            "SIE"=parse(text = TeX("$P_{SIE}$")),
                            "STJ"=parse(text = TeX("$P_{STJ}$")),
                            "TAM"=parse(text = TeX("$P_{TAM}$")),
                            "VAL"=parse(text = TeX("$P_{VAL}$")),
                            "VER"=parse(text = TeX("$P_{VER}$")))) +
   scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Mean"),
                        name = NULL) +
  scale_fill_manual(values=c("orangered3",
                             "gold2","darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Q1: Northern Africa", 
                                                   "Q2: Corsica",
                                                   "Q3: Central Spain",
                                                   "Q4: French Atlantic",
                                                   "Q5: Iberian Atlantic",
                                                   "Q6: South-eastern Spain"),name="Gene pools") +
  labs(title = "Asturias",
       y        = "", 
       x        = TeX("Intercepts P_{p,Asturias}")
       )+
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
```

### Merging the plots

In the Supplementary Information.

```{r M2provIntercepts,fig.height=7,fig.width=15,warning=F,message=F}
pp2_all <- pm2_all + theme(legend.position = "none")
pp2_asturias <- pm2_asturias + theme(legend.position = "none")
pp2_bordeaux <- pm2_bordeaux + theme(legend.position = "none")
pp2_caceres <- pm2_caceres + theme(legend.position = "none")
pp2_portugal <- pm2_portugal + theme(legend.position = "none")
pp2_madrid <- pm2_madrid + theme(legend.position = "none")

g <-  ggarrange(pp2_all,pp2_asturias,pp2_bordeaux,pp2_caceres,pp2_portugal,pp2_madrid ,
                nrow=1)
g
ggsave(g,file="../../figs/SuppInfo/M2_SiteProvIntercepts.png",width=20,height=12) 
```



# **Model M3**


```{r loadM3}
mod <- readRDS(file="../../outputs/models/P1/MOD3.rds")
```


## Main parameters

### Standard deviations

#### Mean

```{r TableM3SDMean}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_site_age__Intercept'='$\\sigma_{cs_{is}}$',
                              'sd_clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Median

```{r TableM3SDMedian}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_site_age__Intercept'='$\\sigma_{cs_{is}}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs


```{r M3MCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
    scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            "sd_site:block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_prov:clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            'sd_site_age__Intercept'=parse(text = TeX("$\\sigma_{cs_{is}}$")),
                            "sd_clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            "sd_prov__Intercept"=parse(text = TeX("$\\sigma_{P}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM3VARMean, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site_age__Intercept'='$\\sigma^{2}_{cs_{is}}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM3VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_site_age__Intercept'='$\\sigma^{2}_{cs_{is}}$',
                              'sd_clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M3_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```



#### Figs

In the Supplementary Information.

```{r M3VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_clon|sd_prov|sd_site:block|sigma)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov:site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{Inter}$")),
                            "square(sd_clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov__Intercept)"=parse(text = TeX("$\\sigma^{2}_{P}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site_)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site_age__Intercept)"=parse(text = TeX("$\\sigma^{2}_{cs_{is}}")),
                            "square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(0.9,0.55,0.55),nrow=3,vjust=c(1.4,1,1))
ggsave(p,file="../../figs/SuppInfo/M3_PostMainVar.png",height=8) 
p
```



## Site intercepts

### Tables


#### Mean

```{r TableM3MeanSiteIntercepts}
df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T,conf.level = 0.95) %>% 
  filter(str_detect(term, "^(r_site\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate,"SD"=Std.error,"InfCI"=Conf.low,"SupCI"=Conf.high) %>% 
  mutate(Parameter = recode_factor(Term,'r_site[asturias,Intercept]'="$S_{Asturias}$",
                              'r_site[bordeaux,Intercept]'= '$S_{Bordeaux}$',
                              'r_site[caceres,Intercept]'='$S_{Caceres}$',
                              'r_site[madrid,Intercept]'='$S_{Madrid}$',
                              'r_site[portugal,Intercept]'="$S_{Portugal}$")) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)
df %>% knitr::kable(digits = 3 ,format = "markdown")
```


#### Median 

```{r TableM3MedianSiteIntercepts}
df <- mod %>% broom::tidyMCMC(estimate.method = "median",conf.int = T,conf.level = 0.95) %>% 
  filter(str_detect(term, "^(r_site\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Median"=Estimate,"SD"=Std.error,"InfCI"=Conf.low,"SupCI"=Conf.high) %>% 
  mutate(Parameter = recode_factor(Term,'r_site[asturias,Intercept]'="$S_{Asturias}$",
                              'r_site[bordeaux,Intercept]'= '$S_{Bordeaux}$',
                              'r_site[caceres,Intercept]'='$S_{Caceres}$',
                              'r_site[madrid,Intercept]'='$S_{Madrid}$',
                              'r_site[portugal,Intercept]'="$S_{Portugal}$")) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)
df %>% knitr::kable(digits = 3 ,format = "markdown")
```


### Figs

#### MCMC intervals

```{r M3SiteInterceptsMCMCintervals, fig.width=8,fig.heifgt=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^r_site\\[",
                    prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() + 
    scale_y_discrete(labels=c('r_site[asturias,Intercept]'=parse(text = TeX("$S_{Asturias}$")),
                              'r_site[bordeaux,Intercept]'=parse(text = TeX("$S_{Bordeaux}$")),
                              'r_site[caceres,Intercept]'=parse(text = TeX("$S_{Caceres}$")),
                              'r_site[madrid,Intercept]'=parse(text = TeX("$S_{Madrid}$")),
                              'r_site[portugal,Intercept]'=parse(text = TeX("$S_{Portugal}$"))
                            )) +
  theme(axis.text = element_text(size=16))
```



## Intercepts of the annual climate in the test sites

### Tables

#### Mean

```{r TableM3SMeanSiteClimSimIntercepts, echo=F}
sclim <- unique(mod$data$site_age)
par <-paste(paste0("'r_site_age[",sclim,",Intercept]'")) 
sclim <-c(str_sub(sclim[1:10],0,-3),str_sub(sclim[11],0,-2))
par2 <-paste(paste0("'$cs_{",c(1,4,3,2,1,1,3,2,1,2,1),",",sclim,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_site_age\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 
                              'r_site_age[asturias10,Intercept]' = '$cs_{1,Asturias}$',
                              'r_site_age[portugal27,Intercept]' = '$cs_{4,Portugal}$',
                              'r_site_age[portugal20,Intercept]' = '$cs_{3,Portugal}$',
                              'r_site_age[asturias21,Intercept]' = '$cs_{2,Asturias}$',
                              'r_site_age[portugal11,Intercept]' = '$cs_{1,Portugal}$',
                              'r_site_age[madrid13,Intercept]' = '$cs_{1,Madrid}$',
                              'r_site_age[asturias37,Intercept]' = '$cs_{3,Asturias}$',
                              'r_site_age[portugal15,Intercept]' = '$cs_{2,Portugal}$',
                              'r_site_age[bordeaux25,Intercept]' = '$cs_{1,Bordeaux}$',
                              'r_site_age[bordeaux37,Intercept]' = '$cs_{2,Bordeaux}$',
                              'r_site_age[caceres8,Intercept]' = '$cs_{1,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

#### Median


In the Supplementary Information.

```{r TableM3MedianSiteClimSimIntercepts}
df <- mod %>% broom::tidyMCMC(estimate.method = "median",conf.int = T,conf.level = 0.95) %>% 
  filter(str_detect(term, "^(r_site_age\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Median"=Estimate,"SD"=Std.error,"InfCI"=Conf.low,"SupCI"=Conf.high) %>% 
  mutate(Parameter = recode_factor(Term,
                              'r_site_age[asturias10,Intercept]' = '$cs_{1,Asturias}$',
                              'r_site_age[portugal27,Intercept]' = '$cs_{4,Portugal}$',
                              'r_site_age[portugal20,Intercept]' = '$cs_{3,Portugal}$',
                              'r_site_age[asturias21,Intercept]' = '$cs_{2,Asturias}$',
                              'r_site_age[portugal11,Intercept]' = '$cs_{1,Portugal}$',
                              'r_site_age[madrid13,Intercept]' = '$cs_{1,Madrid}$',
                              'r_site_age[asturias37,Intercept]' = '$cs_{3,Asturias}$',
                              'r_site_age[portugal15,Intercept]' = '$cs_{2,Portugal}$',
                              'r_site_age[bordeaux25,Intercept]' = '$cs_{1,Bordeaux}$',
                              'r_site_age[bordeaux37,Intercept]' = '$cs_{2,Bordeaux}$',
                              'r_site_age[caceres8,Intercept]' = '$cs_{1,Caceres}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)
df %>% knitr::kable(digits = 3 ,format = "markdown")

print(xtable(df, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M3_SiteClimSimInterceptsPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```


### Figs


In the Supplementary Information.


```{r M3SiteClimSimIntercepts, fig.height=5,fig.width=10,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_site_age\\[")
colnames(POST) <- str_sub(colnames(POST),12,-12) %>% str_to_title()
POST <- as.data.frame(t(POST))
POST$sites <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -sites)

sum <- posteriorsimpelmodellong  %>% 
   group_by(sites) %>%
   dplyr::summarize(mean=mean(value)) 

p <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = sites,
                                    fill   = as.factor(sites),
                                    vline_color = ..quantile..),
                                scale = 1.8, 
                                alpha = .8,
                                size=0.8,
                                rel_min_height=.001,
                                quantile_lines = TRUE, 
                                quantiles = c(0.025,0.5,0.975)) +
  scale_y_discrete(labels=c("Caceres8"=parse(text = TeX("$cs_{1,Caceres}$")),
                            "Bordeaux25"=parse(text = TeX("$cs_{1,Bordeaux}$")),
                            "Bordeaux37"=parse(text = TeX("$cs_{2,Bordeaux}$")),
                            "Portugal11"=parse(text = TeX("$cs_{1,Portugal}$")),
                            "Portugal15"=parse(text = TeX("$cs_{2,Portugal}$")),
                            "Portugal20"=parse(text = TeX("$cs_{3,Portugal}$")),
                            "Portugal27"=parse(text = TeX("$cs_{4,Portugal}$")),
                            "Madrid13"=parse(text = TeX("$cs_{1,Madrid}$")),
                            "Asturias10"=parse(text = TeX("$cs_{1,Asturias}$")),
                            "Asturias21"=parse(text = TeX("$cs_{2,Asturias}$")),
                            "Asturias37"=parse(text = TeX("$cs_{3,Asturias}$"))
                            )) +
  # coord_cartesian(c(-,0.8))+
  scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("5% & 95% quantiles", "mean"),
                        name = NULL) +
  labs(title="",
       y        = "", 
       x = TeX("Estimate of parameter $cs_{is}$")) +
  scale_fill_manual(values=c(vir_lite("cyan2",ds=ds),
                             vir_lite("cyan2",ds=ds),
                             vir_lite("cyan2",ds=ds),
                             vir_lite("navyblue",ds=ds),
                             vir_lite("navyblue",ds=ds),
                             vir_lite("pink",ds=ds),
                             vir_lite("deeppink",ds=ds),
                             vir_lite("dodgerblue2",ds=ds),
                             vir_lite("dodgerblue2",ds=ds),
                             vir_lite("dodgerblue2",ds=ds),
                             vir_lite("dodgerblue2",ds=ds)
                             )) +
  theme_bw() + theme(axis.text = element_text(size=22),axis.title = element_text(size=22), 
                     legend.position = "none",plot.title = element_text(size=22))

p
ggsave(p,file="../../figs/SuppInfo/M3_SiteClimSimInterceptsPost.png",height=7,width=12)
```

## Provenance intercepts

### Table 

```{r TableM3MeanProvIntercepts}
provs <- unique(data$prov)
par <-paste(paste0("'r_prov[",provs,",Intercept]'")) 
par2 <-paste(paste0("'$P_{",provs,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_prov\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_prov[MIM,Intercept]' = '$P_{MIM}$','r_prov[CEN,Intercept]' = '$P_{CEN}$','r_prov[ORI,Intercept]' = '$P_{ORI}$','r_prov[STJ,Intercept]' = '$P_{STJ}$','r_prov[HOU,Intercept]' = '$P_{HOU}$','r_prov[CUE,Intercept]' = '$P_{CUE}$','r_prov[TAM,Intercept]' = '$P_{TAM}$','r_prov[LEI,Intercept]' = '$P_{LEI}$','r_prov[VER,Intercept]' = '$P_{VER}$','r_prov[CAS,Intercept]' = '$P_{CAS}$','r_prov[PIA,Intercept]' = '$P_{PIA}$','r_prov[ARM,Intercept]' = '$P_{ARM}$','r_prov[PET,Intercept]' = '$P_{PET}$','r_prov[VAL,Intercept]' = '$P_{VAL}$','r_prov[SAL,Intercept]' = '$P_{SAL}$','r_prov[OLO,Intercept]' = '$P_{OLO}$','r_prov[CAD,Intercept]' = '$P_{CAD}$','r_prov[ARN,Intercept]' = '$P_{ARN}$','r_prov[BAY,Intercept]' = '$P_{BAY}$','r_prov[SIE,Intercept]' = '$P_{SIE}$','r_prov[SEG,Intercept]' = '$P_{SEG}$','r_prov[PLE,Intercept]' = '$P_{PLE}$','r_prov[BON,Intercept]' = '$P_{BON}$','r_prov[COC,Intercept]' = '$P_{COC}$','r_prov[SAC,Intercept]' = '$P_{SAC}$','r_prov[QUA,Intercept]' = '$P_{QUA}$','r_prov[CAR,Intercept]' = '$P_{CAR}$','r_prov[OLB,Intercept]' = '$P_{OLB}$','r_prov[PIE,Intercept]' = '$P_{PIE}$','r_prov[PUE,Intercept]' = '$P_{PUE}$','r_prov[ALT,Intercept]' = '$P_{ALT}$','r_prov[LAM,Intercept]' = '$P_{LAM}$','r_prov[MAD,Intercept]' = '$P_{MAD}$','r_prov[COM,Intercept]' = '$P_{COM}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M3PostProvIntercepts, fig.width=10,fig.height=8, warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov\\[")
colnames(POST) <- str_sub(colnames(POST),8,-12)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- droplevels(data)
ps <- data %>%
  group_by(prov) %>% 
  summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

pM3_prov <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov),
                                    vline_color = ..quantile..),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=c(.01),
                                size=0.2,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$P_{ALT}$")),
                            "ARM"=parse(text = TeX("$P_{ARM}$")),
                            "ARN"=parse(text = TeX("$P_{ARN}$")),
                            "BAY"=parse(text = TeX("$P_{BAY}$")),
                            "BON"=parse(text = TeX("$P_{BON}$")),
                            "CAD"=parse(text = TeX("$P_{CAD}$")),
                            "CAR"=parse(text = TeX("$P_{CAR}$")),
                            "CAS"=parse(text = TeX("$P_{CAS}$")),
                            "CEN"=parse(text = TeX("$P_{CEN}$")),
                            "COC"=parse(text = TeX("$P_{COC}$")),
                            "COM"=parse(text = TeX("$P_{COM}$")),
                            "CUE"=parse(text = TeX("$P_{CUE}$")),
                            "HOU"=parse(text = TeX("$P_{HOU}$")),
                            "LAM"=parse(text = TeX("$P_{LAM}$")),
                            "LEI"=parse(text = TeX("$P_{LEI}$")),
                            "MAD"=parse(text = TeX("$P_{MAD}$")),
                            "MIM"=parse(text = TeX("$P_{MIM}$")),
                            "OLB"=parse(text = TeX("$P_{OLB}$")),
                            "OLO"=parse(text = TeX("$P_{OLO}$")),
                            "ORI"=parse(text = TeX("$P_{ORI}$")),
                            "PET"=parse(text = TeX("$P_{PET}$")),
                            "PIA"=parse(text = TeX("$P_{PIA}$")),
                            "PIE"=parse(text = TeX("$P_{PIE}$")),
                            "PLE"=parse(text = TeX("$P_{PLE}$")),
                            "PUE"=parse(text = TeX("$P_{PUE}$")),
                            "QUA"=parse(text = TeX("$P_{QUA}$")),
                            "SAC"=parse(text = TeX("$P_{SAC}$")),
                            "SAL"=parse(text = TeX("$P_{SAL}$")),
                            "SEG"=parse(text = TeX("$P_{SEG}$")),
                            "SIE"=parse(text = TeX("$P_{SIE}$")),
                            "STJ"=parse(text = TeX("$P_{STJ}$")),
                            "TAM"=parse(text = TeX("$P_{TAM}$")),
                            "VAL"=parse(text = TeX("$P_{VAL}$")),
                            "VER"=parse(text = TeX("$P_{VER}$")))) +
  
   coord_cartesian(c(-0.35,0.25))+
  
   scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Median"),
                        name = NULL) +
  scale_fill_manual(values=c("orangered3",
                             "gold2","darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Northern Africa (NA)", 
                                                   "Corsica (C)",
                                                   "Central Spain (CS)",
                                                   "French Atlantic (FA)",
                                                   "Iberian Atlantic (IA)",
                                                   "South-eastern Spain (SES)")) +
  labs(fill     = "Gene pools",
       y        = "", 
       x        = TeX("Provenance intercept P_{p}")
       )+
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=15),legend.title = element_text(size=17))# + theme(legend.position = c(0.2,0.75),legend.key = element_blank(),legend.background=element_blank()) 

pM3_prov
```




# **Model M3bis**


```{r loadM3bis}
mod <- readRDS(file="../../outputs/models/P1/MOD13.rds")
```


## Main parameters

### Standard deviations

#### Mean

```{r TableM3bisSDMean}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_site_age__Intercept'='$\\sigma_{cs_{is}}$',
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Median

```{r TableM3bisSDMedian}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_site_age__Intercept'='$\\sigma_{cs_{is}}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs


```{r M3bisMCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
      scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            "sd_site:block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_prov:clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            'sd_site_age__Intercept'=parse(text = TeX("$\\sigma_{cs_{is}}$")),
                            "sd_clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            "sd_prov__Intercept"=parse(text = TeX("$\\sigma_{P}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM3bisVARMean, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site_age__Intercept'='$\\sigma^{2}_{cs_{is}}$',
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM3bisVARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_site_age__Intercept'='$\\sigma^{2}_{cs_{is}}$',
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M3bis_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```



#### Figs

In the Supplementary Information.


```{r M3bisVARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_clon|sd_prov|sd_block|sigma)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov:site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{Inter}$")),
                            "square(sd_clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov__Intercept)"=parse(text = TeX("$\\sigma^{2}_{P}$")),
                            "square(sd_block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site_)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site_age__Intercept)"=parse(text = TeX("$\\sigma^{2}_{cs_{is}}")),
                            "square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(0.9,0.55,0.55),nrow=3,vjust=c(1.4,1,1))
ggsave(p,file="../../figs/SuppInfo/M3bis_PostMainVar.png",height=8) 
p
```


## Intercepts of the annual climate in the test sites

### Tables

#### Mean

```{r TableM3bisSMeanSiteClimSimIntercepts, echo=F}
sclim <- unique(mod$data$site_age)
par <-paste(paste0("'r_site_age[",sclim,",Intercept]'")) 
sclim <-c(str_sub(sclim[1:10],0,-3),str_sub(sclim[11],0,-2))
par2 <-paste(paste0("'$cs_{",c(1,4,3,2,1,1,3,2,1,2,1),",",sclim,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_site_age\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 
                              'r_site_age[asturias10,Intercept]' = '$cs_{1,Asturias}$',
                              'r_site_age[portugal27,Intercept]' = '$cs_{4,Portugal}$',
                              'r_site_age[portugal20,Intercept]' = '$cs_{3,Portugal}$',
                              'r_site_age[asturias21,Intercept]' = '$cs_{2,Asturias}$',
                              'r_site_age[portugal11,Intercept]' = '$cs_{1,Portugal}$',
                              'r_site_age[madrid13,Intercept]' = '$cs_{1,Madrid}$',
                              'r_site_age[asturias37,Intercept]' = '$cs_{3,Asturias}$',
                              'r_site_age[portugal15,Intercept]' = '$cs_{2,Portugal}$',
                              'r_site_age[bordeaux25,Intercept]' = '$cs_{1,Bordeaux}$',
                              'r_site_age[bordeaux37,Intercept]' = '$cs_{2,Bordeaux}$',
                              'r_site_age[caceres8,Intercept]' = '$cs_{1,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

#### Median


In the Supplementary Information.

```{r TableM3bisMedianSiteClimSimIntercepts}
df <- mod %>% broom::tidyMCMC(estimate.method = "median",conf.int = T,conf.level = 0.95) %>% 
  filter(str_detect(term, "^(r_site_age\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Median"=Estimate,"SD"=Std.error,"InfCI"=Conf.low,"SupCI"=Conf.high) %>% 
  mutate(Parameter = recode_factor(Term,
                              'r_site_age[asturias10,Intercept]' = '$cs_{1,Asturias}$',
                              'r_site_age[portugal27,Intercept]' = '$cs_{4,Portugal}$',
                              'r_site_age[portugal20,Intercept]' = '$cs_{3,Portugal}$',
                              'r_site_age[asturias21,Intercept]' = '$cs_{2,Asturias}$',
                              'r_site_age[portugal11,Intercept]' = '$cs_{1,Portugal}$',
                              'r_site_age[madrid13,Intercept]' = '$cs_{1,Madrid}$',
                              'r_site_age[asturias37,Intercept]' = '$cs_{3,Asturias}$',
                              'r_site_age[portugal15,Intercept]' = '$cs_{2,Portugal}$',
                              'r_site_age[bordeaux25,Intercept]' = '$cs_{1,Bordeaux}$',
                              'r_site_age[bordeaux37,Intercept]' = '$cs_{2,Bordeaux}$',
                              'r_site_age[caceres8,Intercept]' = '$cs_{1,Caceres}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)
df %>% knitr::kable(digits = 3 ,format = "markdown")

print(xtable(df, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M3bis_SiteClimSimInterceptsPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```


### Figs


In the Supplementary Information.


```{r M3bisSiteClimSimIntercepts, fig.height=5,fig.width=10,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_site_age\\[")
colnames(POST) <- str_sub(colnames(POST),12,-12) %>% str_to_title()
POST <- as.data.frame(t(POST))
POST$sites <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -sites)

sum <- posteriorsimpelmodellong  %>% 
   group_by(sites) %>%
   dplyr::summarize(mean=mean(value)) 

p <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = sites,
                                    fill   = as.factor(sites),
                                    vline_color = ..quantile..),
                                scale = 1.8, 
                                alpha = .8,
                                size=0.8,
                                rel_min_height=.001,
                                quantile_lines = TRUE, 
                                quantiles = c(0.025,0.5,0.975)) +
  scale_y_discrete(labels=c("Caceres8"=parse(text = TeX("$cs_{1,Caceres}$")),
                            "Bordeaux25"=parse(text = TeX("$cs_{1,Bordeaux}$")),
                            "Bordeaux37"=parse(text = TeX("$cs_{2,Bordeaux}$")),
                            "Portugal11"=parse(text = TeX("$cs_{1,Portugal}$")),
                            "Portugal15"=parse(text = TeX("$cs_{2,Portugal}$")),
                            "Portugal20"=parse(text = TeX("$cs_{3,Portugal}$")),
                            "Portugal27"=parse(text = TeX("$cs_{4,Portugal}$")),
                            "Madrid13"=parse(text = TeX("$cs_{1,Madrid}$")),
                            "Asturias10"=parse(text = TeX("$cs_{1,Asturias}$")),
                            "Asturias21"=parse(text = TeX("$cs_{2,Asturias}$")),
                            "Asturias37"=parse(text = TeX("$cs_{3,Asturias}$"))
                            )) +
  # coord_cartesian(c(-,0.8))+
  scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("5% & 95% quantiles", "mean"),
                        name = NULL) +
  labs(title="",
       y        = "", 
       x = TeX("Estimate of parameter $cs_{is}$")) +
  scale_fill_manual(values=c(vir_lite("cyan2",ds=ds),
                             vir_lite("cyan2",ds=ds),
                             vir_lite("cyan2",ds=ds),
                             vir_lite("navyblue",ds=ds),
                             vir_lite("navyblue",ds=ds),
                             vir_lite("pink",ds=ds),
                             vir_lite("deeppink",ds=ds),
                             vir_lite("dodgerblue2",ds=ds),
                             vir_lite("dodgerblue2",ds=ds),
                             vir_lite("dodgerblue2",ds=ds),
                             vir_lite("dodgerblue2",ds=ds)
                             )) +
  theme_bw() + theme(axis.text = element_text(size=22),axis.title = element_text(size=22), 
                     legend.position = "none",plot.title = element_text(size=22))

p
ggsave(p,file="../../figs/SuppInfo/M3bis_SiteClimSimInterceptsPost.png",height=7,width=12)
```

## Provenance intercepts

### Table 

```{r TableM3bisMeanProvIntercepts}
provs <- unique(data$prov)
par <-paste(paste0("'r_prov[",provs,",Intercept]'")) 
par2 <-paste(paste0("'$P_{",provs,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_prov\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_prov[MIM,Intercept]' = '$P_{MIM}$','r_prov[CEN,Intercept]' = '$P_{CEN}$','r_prov[ORI,Intercept]' = '$P_{ORI}$','r_prov[STJ,Intercept]' = '$P_{STJ}$','r_prov[HOU,Intercept]' = '$P_{HOU}$','r_prov[CUE,Intercept]' = '$P_{CUE}$','r_prov[TAM,Intercept]' = '$P_{TAM}$','r_prov[LEI,Intercept]' = '$P_{LEI}$','r_prov[VER,Intercept]' = '$P_{VER}$','r_prov[CAS,Intercept]' = '$P_{CAS}$','r_prov[PIA,Intercept]' = '$P_{PIA}$','r_prov[ARM,Intercept]' = '$P_{ARM}$','r_prov[PET,Intercept]' = '$P_{PET}$','r_prov[VAL,Intercept]' = '$P_{VAL}$','r_prov[SAL,Intercept]' = '$P_{SAL}$','r_prov[OLO,Intercept]' = '$P_{OLO}$','r_prov[CAD,Intercept]' = '$P_{CAD}$','r_prov[ARN,Intercept]' = '$P_{ARN}$','r_prov[BAY,Intercept]' = '$P_{BAY}$','r_prov[SIE,Intercept]' = '$P_{SIE}$','r_prov[SEG,Intercept]' = '$P_{SEG}$','r_prov[PLE,Intercept]' = '$P_{PLE}$','r_prov[BON,Intercept]' = '$P_{BON}$','r_prov[COC,Intercept]' = '$P_{COC}$','r_prov[SAC,Intercept]' = '$P_{SAC}$','r_prov[QUA,Intercept]' = '$P_{QUA}$','r_prov[CAR,Intercept]' = '$P_{CAR}$','r_prov[OLB,Intercept]' = '$P_{OLB}$','r_prov[PIE,Intercept]' = '$P_{PIE}$','r_prov[PUE,Intercept]' = '$P_{PUE}$','r_prov[ALT,Intercept]' = '$P_{ALT}$','r_prov[LAM,Intercept]' = '$P_{LAM}$','r_prov[MAD,Intercept]' = '$P_{MAD}$','r_prov[COM,Intercept]' = '$P_{COM}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M3bisPostProvIntercepts, fig.width=10,fig.height=8, warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov\\[")
colnames(POST) <- str_sub(colnames(POST),8,-12)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- droplevels(data)
ps <- data %>%
  group_by(prov) %>% 
  summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

pM3bis_prov <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov),
                                    vline_color = ..quantile..),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=c(.01),
                                size=0.2,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$P_{ALT}$")),
                            "ARM"=parse(text = TeX("$P_{ARM}$")),
                            "ARN"=parse(text = TeX("$P_{ARN}$")),
                            "BAY"=parse(text = TeX("$P_{BAY}$")),
                            "BON"=parse(text = TeX("$P_{BON}$")),
                            "CAD"=parse(text = TeX("$P_{CAD}$")),
                            "CAR"=parse(text = TeX("$P_{CAR}$")),
                            "CAS"=parse(text = TeX("$P_{CAS}$")),
                            "CEN"=parse(text = TeX("$P_{CEN}$")),
                            "COC"=parse(text = TeX("$P_{COC}$")),
                            "COM"=parse(text = TeX("$P_{COM}$")),
                            "CUE"=parse(text = TeX("$P_{CUE}$")),
                            "HOU"=parse(text = TeX("$P_{HOU}$")),
                            "LAM"=parse(text = TeX("$P_{LAM}$")),
                            "LEI"=parse(text = TeX("$P_{LEI}$")),
                            "MAD"=parse(text = TeX("$P_{MAD}$")),
                            "MIM"=parse(text = TeX("$P_{MIM}$")),
                            "OLB"=parse(text = TeX("$P_{OLB}$")),
                            "OLO"=parse(text = TeX("$P_{OLO}$")),
                            "ORI"=parse(text = TeX("$P_{ORI}$")),
                            "PET"=parse(text = TeX("$P_{PET}$")),
                            "PIA"=parse(text = TeX("$P_{PIA}$")),
                            "PIE"=parse(text = TeX("$P_{PIE}$")),
                            "PLE"=parse(text = TeX("$P_{PLE}$")),
                            "PUE"=parse(text = TeX("$P_{PUE}$")),
                            "QUA"=parse(text = TeX("$P_{QUA}$")),
                            "SAC"=parse(text = TeX("$P_{SAC}$")),
                            "SAL"=parse(text = TeX("$P_{SAL}$")),
                            "SEG"=parse(text = TeX("$P_{SEG}$")),
                            "SIE"=parse(text = TeX("$P_{SIE}$")),
                            "STJ"=parse(text = TeX("$P_{STJ}$")),
                            "TAM"=parse(text = TeX("$P_{TAM}$")),
                            "VAL"=parse(text = TeX("$P_{VAL}$")),
                            "VER"=parse(text = TeX("$P_{VER}$")))) +
  
   coord_cartesian(c(-0.35,0.25))+
  
   scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Median"),
                        name = NULL) +
  scale_fill_manual(values=c("orangered3",
                             "gold2","darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Northern Africa (NA)", 
                                                   "Corsica (C)",
                                                   "Central Spain (CS)",
                                                   "French Atlantic (FA)",
                                                   "Iberian Atlantic (IA)",
                                                   "South-eastern Spain (SES)")) +
  labs(fill     = "Gene pools",
       y        = "", 
       x        = TeX("Provenance intercept P_{p}")
       )+
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=15),legend.title = element_text(size=17))# + theme(legend.position = c(0.2,0.75),legend.key = element_blank(),legend.background=element_blank()) 

pM3bis_prov
```


# **Model M4**


```{r loadM4}
mod <- readRDS(file="../../outputs/models/P1/MOD4.rds")
```


## Main parameters

### Standard deviations

#### Mean

```{r TableM4SDMean}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_site_age__Intercept'='$\\sigma_{cs_{is}}$',
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Median

```{r TableM4SDMedian}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_site_age__Intercept'='$\\sigma_{cs_{is}}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs

```{r M4MCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            "sd_site:block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_prov:clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            'sd_site_age__Intercept'=parse(text = TeX("$\\sigma_{cs_{is}}$")),
                            "sd_clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            "sd_prov__Intercept"=parse(text = TeX("$\\sigma_{P}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'=parse(text = TeX("$\\sigma_{g_{j}}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM4VARMean, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site_age__Intercept'='$\\sigma^{2}_{cs_{is}}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM4VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_site_age__Intercept'='$\\sigma^{2}_{cs_{is}}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M4_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```



#### Figs

In the Supplementary Information.

```{r M4VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_clon|sd_prov|sd_site:block|sigma|sd_mm)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov:site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{Inter}$")),
                            "square(sd_mmQ1Q2Q3Q4Q5Q6__Intercept)"=parse(text = TeX("$\\sigma^{2}_{g_{j}}$")),
                            "square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov__Intercept)"=parse(text = TeX("$\\sigma^{2}_{P}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site_)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site_age__Intercept)"=parse(text = TeX("$\\sigma^{2}_{cs_{is}}")),
                            "square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(0.9,0.55,0.55),nrow=3,vjust=c(1.2,1,1))
ggsave(p,file="../../figs/SuppInfo/M4_PostMainVar.png",height=8) 
p
```



## Provenance intercepts

### Table 

```{r TableM4MeanProvIntercepts}
provs <- unique(data$prov)
par <-paste(paste0("'r_prov[",provs,",Intercept]'")) 
par2 <-paste(paste0("'$P_{",provs,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_prov\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_prov[MIM,Intercept]' = '$P_{MIM}$','r_prov[CEN,Intercept]' = '$P_{CEN}$','r_prov[ORI,Intercept]' = '$P_{ORI}$','r_prov[STJ,Intercept]' = '$P_{STJ}$','r_prov[HOU,Intercept]' = '$P_{HOU}$','r_prov[CUE,Intercept]' = '$P_{CUE}$','r_prov[TAM,Intercept]' = '$P_{TAM}$','r_prov[LEI,Intercept]' = '$P_{LEI}$','r_prov[VER,Intercept]' = '$P_{VER}$','r_prov[CAS,Intercept]' = '$P_{CAS}$','r_prov[PIA,Intercept]' = '$P_{PIA}$','r_prov[ARM,Intercept]' = '$P_{ARM}$','r_prov[PET,Intercept]' = '$P_{PET}$','r_prov[VAL,Intercept]' = '$P_{VAL}$','r_prov[SAL,Intercept]' = '$P_{SAL}$','r_prov[OLO,Intercept]' = '$P_{OLO}$','r_prov[CAD,Intercept]' = '$P_{CAD}$','r_prov[ARN,Intercept]' = '$P_{ARN}$','r_prov[BAY,Intercept]' = '$P_{BAY}$','r_prov[SIE,Intercept]' = '$P_{SIE}$','r_prov[SEG,Intercept]' = '$P_{SEG}$','r_prov[PLE,Intercept]' = '$P_{PLE}$','r_prov[BON,Intercept]' = '$P_{BON}$','r_prov[COC,Intercept]' = '$P_{COC}$','r_prov[SAC,Intercept]' = '$P_{SAC}$','r_prov[QUA,Intercept]' = '$P_{QUA}$','r_prov[CAR,Intercept]' = '$P_{CAR}$','r_prov[OLB,Intercept]' = '$P_{OLB}$','r_prov[PIE,Intercept]' = '$P_{PIE}$','r_prov[PUE,Intercept]' = '$P_{PUE}$','r_prov[ALT,Intercept]' = '$P_{ALT}$','r_prov[LAM,Intercept]' = '$P_{LAM}$','r_prov[MAD,Intercept]' = '$P_{MAD}$','r_prov[COM,Intercept]' = '$P_{COM}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs

```{r M4PostProvs, warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov\\[")
colnames(POST) <- str_sub(colnames(POST),8,-12)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- droplevels(data)
ps <- data %>%
  group_by(prov) %>% 
  summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

pM4_prov <- ggplot() +
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov),
                                    vline_color = ..quantile..),
                                scale = 2, 
                                alpha = .6,
                                rel_min_height=c(.01),
                                size=0.3,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$P_{ALT}$")),
                            "ARM"=parse(text = TeX("$P_{ARM}$")),
                            "ARN"=parse(text = TeX("$P_{ARN}$")),
                            "BAY"=parse(text = TeX("$P_{BAY}$")),
                            "BON"=parse(text = TeX("$P_{BON}$")),
                            "CAD"=parse(text = TeX("$P_{CAD}$")),
                            "CAR"=parse(text = TeX("$P_{CAR}$")),
                            "CAS"=parse(text = TeX("$P_{CAS}$")),
                            "CEN"=parse(text = TeX("$P_{CEN}$")),
                            "COC"=parse(text = TeX("$P_{COC}$")),
                            "COM"=parse(text = TeX("$P_{COM}$")),
                            "CUE"=parse(text = TeX("$P_{CUE}$")),
                            "HOU"=parse(text = TeX("$P_{HOU}$")),
                            "LAM"=parse(text = TeX("$P_{LAM}$")),
                            "LEI"=parse(text = TeX("$P_{LEI}$")),
                            "MAD"=parse(text = TeX("$P_{MAD}$")),
                            "MIM"=parse(text = TeX("$P_{MIM}$")),
                            "OLB"=parse(text = TeX("$P_{OLB}$")),
                            "OLO"=parse(text = TeX("$P_{OLO}$")),
                            "ORI"=parse(text = TeX("$P_{ORI}$")),
                            "PET"=parse(text = TeX("$P_{PET}$")),
                            "PIA"=parse(text = TeX("$P_{PIA}$")),
                            "PIE"=parse(text = TeX("$P_{PIE}$")),
                            "PLE"=parse(text = TeX("$P_{PLE}$")),
                            "PUE"=parse(text = TeX("$P_{PUE}$")),
                            "QUA"=parse(text = TeX("$P_{QUA}$")),
                            "SAC"=parse(text = TeX("$P_{SAC}$")),
                            "SAL"=parse(text = TeX("$P_{SAL}$")),
                            "SEG"=parse(text = TeX("$P_{SEG}$")),
                            "SIE"=parse(text = TeX("$P_{SIE}$")),
                            "STJ"=parse(text = TeX("$P_{STJ}$")),
                            "TAM"=parse(text = TeX("$P_{TAM}$")),
                            "VAL"=parse(text = TeX("$P_{VAL}$")),
                            "VER"=parse(text = TeX("$P_{VER}$")))) +
  
   coord_cartesian(c(-0.35,0.25))+
  
   scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Median"),
                        name = NULL) +
  
  scale_fill_manual(values=c("orangered3",
                             "gold2",
                             "darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Northern Africa", 
                                                   "Corsica",
                                                   "Central Spain",
                                                   "French Atlantic",
                                                   "Iberian Atlantic",
                                                   "South-eastern Spain"),name="Gene pools") +
  labs(fill     = "Gene pools",
       y        = "", 
       x        = TeX("Provenance intercept P_{p}")
       ) + 
  theme_bw() + 
  theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
```

## Gene pool intercepts

### Table

```{r TableM4GenePoolIntercepts}
# gp <- str_c(rep("Q",6),1:6)
# par <-paste(paste0("'r_mmQ1Q2Q3Q4Q5Q6[",gp,",Intercept]'")) 
# par2 <-paste(paste0("'$g_{",1:6,"}$'")) 
# tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_mm)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_mmQ1Q2Q3Q4Q5Q6[Q1,Intercept]' = '$g_{NA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q2,Intercept]' = '$g_{C}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q3,Intercept]' = '$g_{CS}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q4,Intercept]' = '$g_{FA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q5,Intercept]' = '$g_{IA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q6,Intercept]' = '$g_{SES}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs 

```{r M4PostGenePoolIntercepts, warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_mmQ1Q2Q3Q4Q5Q6\\[")
colnames(POST) <- str_sub(colnames(POST),18,-12)
POST <- as.data.frame(t(POST))
POST$genepool <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -genepool)%>%
  group_by(genepool) %>%
  dplyr::mutate(meanpergenepool = mean(value))%>%
  ungroup()

pM4_GP <- ggplot()+
  geom_vline(xintercept = 0, 
             col        = "grey70") +
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(genepool), meanpergenepool),
                                    # height = ..density.., 
                                    fill   = as.factor(genepool),
                                    vline_color = ..quantile..),
                                scale = 2, 
                                alpha = .6,
                                rel_min_height=c(.001),
                                size=0.5,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
        scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Median"),
                        name = NULL) +
  coord_cartesian(c(-0.5,0.4))+
  scale_fill_manual(values=c("orangered3", # html code #CD3700
                             "gold2", # html code #EEC900
                             "darkorchid3",	#9A32CD
                             "navyblue", #000080
                             "turquoise2", #00E5EE
                             "green3" # 
                             ), labels = c("Northern Africa", 
                                                   "Corsica",
                                                   "Central Spain",
                                                   "French Atlantic",
                                                   "Iberian Atlantic",
                                                   "South-eastern Spain")) +
  
  
  scale_y_discrete(labels=c("Q1"=parse(text = TeX("$g_{NA}$")),
                            "Q2"=parse(text = TeX("$g_{C}$")),
                            "Q3"=parse(text = TeX("$g_{CS}$")),
                            "Q4"=parse(text = TeX("$g_{FA}$")),
                            "Q5"=parse(text = TeX("$g_{IA}$")),
                            "Q6"=parse(text = TeX("$g_{SES}$")))) + 
  labs(fill     = "Gene pools",
       y        = "", 
       x        = TeX("Gene pool intercept g_{j}")
       ) + 
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
```



# **Model M5**


```{r loadM5}
mod <- readRDS(file="../../outputs/models/P1/MOD5.rds")
```


## Main parameters

### Standard deviations

#### Mean

```{r TableM5SDMean}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_site_age__Intercept'='$\\sigma_{cs_{is}}$',
                              'sd_clon1__Intercept'='$\\sigma_{A_{NA}}$',
                              'sd_clon2__Intercept'='$\\sigma_{A_{C}}$',
                              'sd_clon3__Intercept'='$\\sigma_{A_{CS}}$',
                              'sd_clon4__Intercept'='$\\sigma_{A_{FA}}$',
                              'sd_clon5__Intercept'='$\\sigma_{A_{IA}}$',
                              'sd_clon6__Intercept'='$\\sigma_{A_{SES}}$',
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Median

```{r TableM5SDMedian}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_site_age__Intercept'='$\\sigma_{cs_{is}}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_clon1__Intercept'='$\\sigma_{A_{NA}}$',
                              'sd_clon2__Intercept'='$\\sigma_{A_{C}}$',
                              'sd_clon3__Intercept'='$\\sigma_{A_{CS}}$',
                              'sd_clon4__Intercept'='$\\sigma_{A_{FA}}$',
                              'sd_clon5__Intercept'='$\\sigma_{A_{IA}}$',
                              'sd_clon6__Intercept'='$\\sigma_{A_{SES}}$',
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs


```{r M5MCMCIntervals, fig.width=10,fig.height=4,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            "sd_site:block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_prov:clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            'sd_site_age__Intercept'=parse(text = TeX("$\\sigma_{cs_{is}}$")),
                            "sd_clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            "sd_prov__Intercept"=parse(text = TeX("$\\sigma_{P}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'=parse(text = TeX("$\\sigma_{g_{j}}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "sd_clon1__Intercept"=parse(text = TeX("$\\sigma_{A_{NA}}$")),
                            "sd_clon2__Intercept"=parse(text = TeX("$\\sigma_{A_{C}}$")),
                            "sd_clon3__Intercept"=parse(text = TeX("$\\sigma_{A_{CS}}$")),
                            "sd_clon4__Intercept"=parse(text = TeX("$\\sigma_{A_{FA}}$")),
                            "sd_clon5__Intercept"=parse(text = TeX("$\\sigma_{A_{IA}}$")),
                            "sd_clon6__Intercept"=parse(text = TeX("$\\sigma_{A_{SES}}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM5VARMean, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site_age__Intercept'='$\\sigma^{2}_{cs_{is}}$',
                              'sd_clon1__Intercept'='$\\sigma^{2}_{A_{NA}}$',
                              'sd_clon2__Intercept'='$\\sigma^{2}_{A_{C}}$',
                              'sd_clon3__Intercept'='$\\sigma^{2}_{A_{CS}}$',
                              'sd_clon4__Intercept'='$\\sigma^{2}_{A_{FA}}$',
                              'sd_clon5__Intercept'='$\\sigma^{2}_{A_{IA}}$',
                              'sd_clon6__Intercept'='$\\sigma^{2}_{A_{SES}}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM5VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_site_age__Intercept'='$\\sigma^{2}_{cs_{is}}$',
                              'sd_clon1__Intercept'='$\\sigma^{2}_{A_{NA}}$',
                              'sd_clon2__Intercept'='$\\sigma^{2}_{A_{C}}$',
                              'sd_clon3__Intercept'='$\\sigma^{2}_{A_{CS}}$',
                              'sd_clon4__Intercept'='$\\sigma^{2}_{A_{FA}}$',
                              'sd_clon5__Intercept'='$\\sigma^{2}_{A_{IA}}$',
                              'sd_clon6__Intercept'='$\\sigma^{2}_{A_{SES}}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M5_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```



#### Figs

In the Supplementary Information.

```{r M5VARFigs,fig.height=8, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_clon|sd_prov|sd_site:block|sigma|sd_mm)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) + coord_cartesian(c(0,0.11)) +
  scale_y_discrete(labels=c("square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov:site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{Inter}$")),
                            "square(sd_mmQ1Q2Q3Q4Q5Q6__Intercept)"=parse(text = TeX("$\\sigma^{2}_{g_{j}}$")),
                            "square(sd_clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov__Intercept)"=parse(text = TeX("$\\sigma^{2}_{P}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$")),
                            'square(sd_clon1__Intercept)'=parse(text = TeX("$\\sigma^{2}_{A_{NA}}$")),
                            'square(sd_clon2__Intercept)'=parse(text = TeX("$\\sigma^{2}_{A_{C}}$")),
                            'square(sd_clon3__Intercept)'=parse(text = TeX("$\\sigma^{2}_{A_{CS}}$")),
                            'square(sd_clon4__Intercept)'=parse(text = TeX("$\\sigma^{2}_{A_{FA}}$")),
                            'square(sd_clon5__Intercept)'=parse(text = TeX("$\\sigma^{2}_{A_{IA}}$")),
                            'square(sd_clon6__Intercept)'=parse(text = TeX("$\\sigma^{2}_{A_{SES}}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site_)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site_age__Intercept)"=parse(text = TeX("$\\sigma^{2}_{cs_{is}}")),
                            "square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(1.4,0.55,0.55),nrow=3,vjust=c(1.2,1,1))
ggsave(p,file="../../figs/SuppInfo/M5_PostMainVar.png",height=8) 
p
```




## Genetic variance

### Total genetic variances

```{r M5AddGenVar, warning=F,message=F}
POST <- posterior_samples(mod,pars = "^sd_clon")
colnames(POST) <- str_sub(colnames(POST),4,-12)
POST <- POST*POST
POST <- as.data.frame(t(POST))
POST$genepool <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -genepool)%>%
  group_by(genepool) %>%
  dplyr::mutate(meanpergenepool = mean(value))%>%
  ungroup()

sum <- posteriorsimpelmodellong  %>% 
   group_by(genepool) %>%
   dplyr::summarize(mean=mean(value)) 

p_sigmaAj <- ggplot()+
  geom_vline(xintercept = 0, col        = "grey70") +
  geom_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(genepool), meanpergenepool),
                                    fill   = as.factor(genepool), vline_color = ..quantile..),
                                scale = 1, 
                                alpha = .6,
                                rel_min_height=c(.0001),
                                size=0.8,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  
  scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("5th & 95th percentiles", "Median"),
                        name = NULL) +
  
  coord_cartesian(c(0,0.1))+
  xlim(0, 1) +
  
  scale_fill_manual(values=c("orangered3","gold2","darkorchid3","navyblue","turquoise2","green3"), 
                    labels = c("Northern Africa (NA)", 
                               "Corsica (C)",
                               "Central Spain (CS)",
                               "French Atlantic (FA)", 
                               "Iberian Atlantic (IA)",
                               "South-eastern Spain (SES)"),
                    name="Gene pools\ncontributing to the\npartial genetic values:") +
  
    scale_y_discrete(labels=c("clon1"=parse(text = TeX("$\\sigma_{A_{NA}}^{2}$")),
                            "clon2"=parse(text = TeX("$\\sigma_{A_{C}}^{2}$")),
                            "clon3"=parse(text = TeX("$\\sigma_{A_{CS}}^{2}$")),
                            "clon4"=parse(text = TeX("$\\sigma_{A_{FA}}^{2}$")),
                            "clon5"=parse(text = TeX("$\\sigma_{A_{IA}}^{2}$")),
                            "clon6"=parse(text = TeX("$\\sigma_{A_{SES}}^{2}$")))) + 
  labs(y  = "", 
       x  = TeX("Variances $\\sigma_{A_{j}}^{2}$ of the partial genetic values $a_{gj}$ from gene pool $j$")) +
  
  theme_bw() + 
  theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20)) 
```



### Partial genetic values

```{r M5PartialGeneticValues, warning=F,message=F}
clon1 <- tidy(mod,parameters="r_clon1", intervals = FALSE)
clon1$clon <- str_sub(clon1$term,9,-12) 
clon1$term <- NULL
colnames(clon1) <- c("est1","std1","clon")

clon2 <- tidy(mod,parameters="r_clon2", intervals = FALSE)
clon2$clon <- str_sub(clon2$term,9,-12) 
clon2$term <- NULL
colnames(clon2) <- c("est2","std2","clon")

clon3 <- tidy(mod,parameters="r_clon3", intervals = FALSE)
clon3$clon <- str_sub(clon3$term,9,-12) 
clon3$term <- NULL
colnames(clon3) <- c("est3","std3","clon")

clon4 <- tidy(mod,parameters="r_clon4", intervals = FALSE)
clon4$clon <- str_sub(clon4$term,9,-12) 
clon4$term <- NULL
colnames(clon4) <- c("est4","std4","clon")

clon5 <- tidy(mod,parameters="r_clon5", intervals = FALSE)
clon5$clon <- str_sub(clon5$term,9,-12) 
clon5$term <- NULL
colnames(clon5) <- c("est5","std5","clon")

clon6 <- tidy(mod,parameters="r_clon6", intervals = FALSE)
clon6$clon <- str_sub(clon6$term,9,-12) 
clon6$term <- NULL
colnames(clon6) <- c("est6","std6","clon")

ps <- data %>% dplyr::select(clon,Q1,Q2,Q3,Q4,Q5,Q6,max.Qvalue,max.Q) %>%  distinct()

g <- Reduce(
  function(x, y, ...) merge(x, y, ...), 
  list(clon1,clon2,clon3,clon4,clon5,clon6,ps)
)

g <- g %>% dplyr::select(starts_with("est"),clon,max.Q) %>%
  pivot_longer(-c(clon,max.Q),names_to="PBV",values_to="est")


pPBV <- ggplot(g, aes(x=max.Q, y=est, fill=PBV)) + 
   geom_boxplot(alpha=0.6) + theme_bw() +
  scale_x_discrete(labels=c("Q1"="Northern Africa",
                            "Q2"="Corsica",
                            "Q3"="Central Spain",
                            "Q4"="French Atlantic",
                            "Q5"="Iberian Atlantic",
                            "Q6"="South-eastern Spain")) + 
  scale_fill_manual(values=c("orangered3","gold2","darkorchid3","navyblue","turquoise2","green3"), 
                    labels = c("Northern Africa", 
                               "Corsica",
                               "Central Spain",
                               "French Atlantic",
                               "Iberian Atlantic",
                               "South-eastern Spain"),
                    name=TeX("Gene pools\ncontributing to the\npartial genetic values:")) + 
  labs(y        = "Means of the posterior distribution\nof the genetic values", 
       x        = TeX("Dominant gene pool for each genotype")) +
  theme(axis.text = element_text(size=12),
        axis.title = element_text(size=14), 
        legend.text = element_text(size=18),
        legend.title = element_text(size=20)) 

```


## Paper figure 


```{r FigureAddGenVarM5, fig.width=10 ,fig.height=9,warning=F,message=F}
pPBVf <- pPBV + theme(legend.position = "none") 
 
pp <-  ggarrange(p_sigmaAj,pPBVf,labels=c("A","B"),nrow=2,heights = c(1, 1.3))
ggsave(pp,file="../../figs/manuscript/TotGenVarM5.png",height=12,width=12)
pp 
```


# **Model M6**


```{r loadM6}
mod <- readRDS(file="../../outputs/models/P1/MOD6.rds")
```


## Main parameters

### Standard deviations

#### Mean

```{r TableM6SDMean}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov_clim__Intercept'='$\\sigma_{cp_{p}}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_site_age__Intercept'='$\\sigma_{cs_{is}}$',
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Median

```{r TableM6SDMedian}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov_clim__Intercept'='$\\sigma_{cp_{p}}$',
                              'sd_site_age__Intercept'='$\\sigma_{cs_{is}}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs

```{r M6MCMCIntervals, fig.width=10,fig.height=4,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            "sd_site:block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_prov:clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            'sd_site_age__Intercept'=parse(text = TeX("$\\sigma_{cs_{is}}$")),
                            "sd_clon__Intercept"=parse(text = TeX("$\\sigma_{G}$")),
                            'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'=parse(text = TeX("$\\sigma_{g_{j}}$")),
                            "sd_prov__Intercept"=parse(text = TeX("$\\sigma_{P}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            'sd_prov_clim__Intercept'=parse(text = TeX("$\\sigma_{cp_{p}}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\sigma_{cp_{p}}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM6VARMean, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov_clim__Intercept'='$\\sigma^{2}_{cp_{p}}$',
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site_age__Intercept'='$\\sigma^{2}_{cs_{is}}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM6VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_site_age__Intercept'='$\\sigma^{2}_{cs_{is}}$',
                              'sd_prov_clim__Intercept'='$\\sigma^{2}_{cp_{p}}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M6_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```



#### Figs

In the Supplementary Information.

```{r M6VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_prov:clon|sd_prov__|sd_site:block|sigma)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) + # coord_cartesian(c(0,0.5)) +
  scale_y_discrete(labels=c("square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov__Intercept)"=parse(text = TeX("$\\sigma^{2}_{P}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 
p4 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_mm|sd_prov_clim)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) + coord_cartesian(c(0,1)) +
  scale_y_discrete(labels=c("square(sd_mmQ1Q2Q3Q4Q5Q6__Intercept)"=parse(text = TeX("$\\sigma^{2}_{g_{j}}$")),
                            "square(sd_prov_clim__Intercept)"=parse(text = TeX("$\\sigma^{2}_{cp_{p}}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site_)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site_age__Intercept)"=parse(text = TeX("$\\sigma^{2}_{cs_{is}}")),
                            "square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p4,p2,p3,labels=c("A","B","C","D"),font.label = list(size=20),heights = c(0.4,0.3,0.3,0.3),nrow=4,vjust=c(1.2,1,1,1))
ggsave(p,file="../../figs/SuppInfo/M6_PostMainVar.png",height=8) 
p
```




## Provenance intercepts


### Table 


```{r TableM6ProvIntercepts}
provs <- unique(data$prov)
par <-paste(paste0("'r_prov[",provs,",Intercept]'")) 
par2 <-paste(paste0("'$P_{",provs,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_prov\\[)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_prov[MIM,Intercept]' = '$P_{MIM}$','r_prov[CEN,Intercept]' = '$P_{CEN}$','r_prov[ORI,Intercept]' = '$P_{ORI}$','r_prov[STJ,Intercept]' = '$P_{STJ}$','r_prov[HOU,Intercept]' = '$P_{HOU}$','r_prov[CUE,Intercept]' = '$P_{CUE}$','r_prov[TAM,Intercept]' = '$P_{TAM}$','r_prov[LEI,Intercept]' = '$P_{LEI}$','r_prov[VER,Intercept]' = '$P_{VER}$','r_prov[CAS,Intercept]' = '$P_{CAS}$','r_prov[PIA,Intercept]' = '$P_{PIA}$','r_prov[ARM,Intercept]' = '$P_{ARM}$','r_prov[PET,Intercept]' = '$P_{PET}$','r_prov[VAL,Intercept]' = '$P_{VAL}$','r_prov[SAL,Intercept]' = '$P_{SAL}$','r_prov[OLO,Intercept]' = '$P_{OLO}$','r_prov[CAD,Intercept]' = '$P_{CAD}$','r_prov[ARN,Intercept]' = '$P_{ARN}$','r_prov[BAY,Intercept]' = '$P_{BAY}$','r_prov[SIE,Intercept]' = '$P_{SIE}$','r_prov[SEG,Intercept]' = '$P_{SEG}$','r_prov[PLE,Intercept]' = '$P_{PLE}$','r_prov[BON,Intercept]' = '$P_{BON}$','r_prov[COC,Intercept]' = '$P_{COC}$','r_prov[SAC,Intercept]' = '$P_{SAC}$','r_prov[QUA,Intercept]' = '$P_{QUA}$','r_prov[CAR,Intercept]' = '$P_{CAR}$','r_prov[OLB,Intercept]' = '$P_{OLB}$','r_prov[PIE,Intercept]' = '$P_{PIE}$','r_prov[PUE,Intercept]' = '$P_{PUE}$','r_prov[ALT,Intercept]' = '$P_{ALT}$','r_prov[LAM,Intercept]' = '$P_{LAM}$','r_prov[MAD,Intercept]' = '$P_{MAD}$','r_prov[COM,Intercept]' = '$P_{COM}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M6PostProvs, warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov\\[")
colnames(POST) <- str_sub(colnames(POST),8,-12)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- droplevels(data)
ps <- data %>%
  group_by(prov) %>% 
  summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()

pM6_prov <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov),
                                    vline_color = ..quantile..),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=c(.01),
                                size=0.3,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  
   coord_cartesian(c(-0.35,0.25))+
  
   scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Median"),
                        name = NULL) +
  

  scale_fill_manual(values=c("orangered3",
                             "gold2",
                             "darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Northern Africa", 
                                                   "Corsica",
                                                   "Central Spain",
                                                   "French Atlantic",
                                                   "Iberian Atlantic",
                                                   "South-eastern Spain")) +
  
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$P_{ALT}$")),
                            "ARM"=parse(text = TeX("$P_{ARM}$")),
                            "ARN"=parse(text = TeX("$P_{ARN}$")),
                            "BAY"=parse(text = TeX("$P_{BAY}$")),
                            "BON"=parse(text = TeX("$P_{BON}$")),
                            "CAD"=parse(text = TeX("$P_{CAD}$")),
                            "CAR"=parse(text = TeX("$P_{CAR}$")),
                            "CAS"=parse(text = TeX("$P_{CAS}$")),
                            "CEN"=parse(text = TeX("$P_{CEN}$")),
                            "COC"=parse(text = TeX("$P_{COC}$")),
                            "COM"=parse(text = TeX("$P_{COM}$")),
                            "CUE"=parse(text = TeX("$P_{CUE}$")),
                            "HOU"=parse(text = TeX("$P_{HOU}$")),
                            "LAM"=parse(text = TeX("$P_{LAM}$")),
                            "LEI"=parse(text = TeX("$P_{LEI}$")),
                            "MAD"=parse(text = TeX("$P_{MAD}$")),
                            "MIM"=parse(text = TeX("$P_{MIM}$")),
                            "OLB"=parse(text = TeX("$P_{OLB}$")),
                            "OLO"=parse(text = TeX("$P_{OLO}$")),
                            "ORI"=parse(text = TeX("$P_{ORI}$")),
                            "PET"=parse(text = TeX("$P_{PET}$")),
                            "PIA"=parse(text = TeX("$P_{PIA}$")),
                            "PIE"=parse(text = TeX("$P_{PIE}$")),
                            "PLE"=parse(text = TeX("$P_{PLE}$")),
                            "PUE"=parse(text = TeX("$P_{PUE}$")),
                            "QUA"=parse(text = TeX("$P_{QUA}$")),
                            "SAC"=parse(text = TeX("$P_{SAC}$")),
                            "SAL"=parse(text = TeX("$P_{SAL}$")),
                            "SEG"=parse(text = TeX("$P_{SEG}$")),
                            "SIE"=parse(text = TeX("$P_{SIE}$")),
                            "STJ"=parse(text = TeX("$P_{STJ}$")),
                            "TAM"=parse(text = TeX("$P_{TAM}$")),
                            "VAL"=parse(text = TeX("$P_{VAL}$")),
                            "VER"=parse(text = TeX("$P_{VER}$")))) +
  labs(fill     = "Gene pools",
       y        = "", 
       x        = TeX("Provenance intercept P_{p}")
       ) + 
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))

```

## Gene pool intercepts

### Table

```{r TableM6GenePoolIntercepts}
gp <- str_c(rep("Q",6),1:6)
par <-paste(paste0("'r_mmQ1Q2Q3Q4Q5Q6[",gp,",Intercept]'")) 
par2 <-paste(paste0("'$g_{",1:6,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_mm)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_mmQ1Q2Q3Q4Q5Q6[Q1,Intercept]' = '$g_{NA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q2,Intercept]' = '$g_{C}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q3,Intercept]' = '$g_{CS}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q4,Intercept]' = '$g_{FA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q5,Intercept]' = '$g_{IA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q6,Intercept]' = '$g_{SES}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M6GenePoolIntercepts,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_mmQ1Q2Q3Q4Q5Q6\\[")
colnames(POST) <- str_sub(colnames(POST),18,-12)
POST <- as.data.frame(t(POST))
POST$genepool <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -genepool)%>%
  group_by(genepool) %>%
  dplyr::mutate(meanpergenepool = mean(value))%>%
  ungroup()

pM6_GP <- ggplot()+
  geom_vline(xintercept = 0, 
             col        = "grey70") +
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(genepool), meanpergenepool),
                                    fill   = as.factor(genepool),
                                    vline_color = ..quantile..),
                                scale = 2, 
                                alpha = .6,
                                rel_min_height=c(.001),
                                size=0.5,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
        scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Median"),
                        name = NULL) +
  coord_cartesian(c(-0.5,0.4))+
  scale_fill_manual(values=c("orangered3",
                             "gold2",
                             "darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Northern Africa", 
                                                   "Corsica",
                                                   "Central Spain",
                                                   "French Atlantic",
                                                   "Iberian Atlantic",
                                                   "South-eastern Spain")) +
  
  
  scale_y_discrete(labels=c("Q1"=parse(text = TeX("$g_{NA}$")),
                            "Q2"=parse(text = TeX("$g_{C}$")),
                            "Q3"=parse(text = TeX("$g_{CS}$")),
                            "Q4"=parse(text = TeX("$g_{FA}$")),
                            "Q5"=parse(text = TeX("$g_{IA}$")),
                            "Q6"=parse(text = TeX("$g_{SES}$")))) + 
  labs(fill     = "Gene pools",
       y        = "", 
       x        = TeX("Gene pool intercept g_{j}")
       ) + 
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
```



## Provenance climate intercepts

### Table

```{r TableM6ProvClimSimIntercepts}
provs <- unique(data$prov)
par <-paste(paste0("'r_prov_clim[",provs,",Intercept]'")) 
par2 <-paste(paste0("'$cp_{",provs,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_prov_clim)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_prov_clim[MIM,Intercept]' = '$cp_{MIM}$','r_prov_clim[CEN,Intercept]' = '$cp_{CEN}$','r_prov_clim[ORI,Intercept]' = '$cp_{ORI}$','r_prov_clim[STJ,Intercept]' = '$cp_{STJ}$','r_prov_clim[HOU,Intercept]' = '$cp_{HOU}$','r_prov_clim[CUE,Intercept]' = '$cp_{CUE}$','r_prov_clim[TAM,Intercept]' = '$cp_{TAM}$','r_prov_clim[LEI,Intercept]' = '$cp_{LEI}$','r_prov_clim[VER,Intercept]' = '$cp_{VER}$','r_prov_clim[CAS,Intercept]' = '$cp_{CAS}$','r_prov_clim[PIA,Intercept]' = '$cp_{PIA}$','r_prov_clim[ARM,Intercept]' = '$cp_{ARM}$','r_prov_clim[PET,Intercept]' = '$cp_{PET}$','r_prov_clim[VAL,Intercept]' = '$cp_{VAL}$','r_prov_clim[SAL,Intercept]' = '$cp_{SAL}$','r_prov_clim[OLO,Intercept]' = '$cp_{OLO}$','r_prov_clim[CAD,Intercept]' = '$cp_{CAD}$','r_prov_clim[ARN,Intercept]' = '$cp_{ARN}$','r_prov_clim[BAY,Intercept]' = '$cp_{BAY}$','r_prov_clim[SIE,Intercept]' = '$cp_{SIE}$','r_prov_clim[SEG,Intercept]' = '$cp_{SEG}$','r_prov_clim[PLE,Intercept]' = '$cp_{PLE}$','r_prov_clim[BON,Intercept]' = '$cp_{BON}$','r_prov_clim[COC,Intercept]' = '$cp_{COC}$','r_prov_clim[SAC,Intercept]' = '$cp_{SAC}$','r_prov_clim[QUA,Intercept]' = '$cp_{QUA}$','r_prov_clim[CAR,Intercept]' = '$cp_{CAR}$','r_prov_clim[OLB,Intercept]' = '$cp_{OLB}$','r_prov_clim[PIE,Intercept]' = '$cp_{PIE}$','r_prov_clim[PUE,Intercept]' = '$cp_{PUE}$','r_prov_clim[ALT,Intercept]' = '$cp_{ALT}$','r_prov_clim[LAM,Intercept]' = '$cp_{LAM}$','r_prov_clim[MAD,Intercept]' = '$cp_{MAD}$','r_prov_clim[COM,Intercept]' = '$cp_{COM}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M6PostProvClimSimIntercepts,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_prov_clim")
colnames(POST) <- str_sub(colnames(POST),13,-12)
POST <- as.data.frame(t(POST))
POST$prov <- as.factor(rownames(POST))

data <- droplevels(data)
ps <- data %>%
  group_by(prov) %>% 
  summarise_at(vars(paste0(rep("Q",6),1:6)), mean)
ps$max.Q.prov <- colnames(ps[,2:7])[apply(ps[,2:7],1,which.max)]
ps$prov <- as.factor(ps$prov)

posteriorsimpelmodellong <- inner_join(POST, ps[,c("prov","max.Q.prov")],by="prov") %>%  as_tibble() %>% 
gather(key = "key", value = "value", -prov,-max.Q.prov)%>%
  group_by(prov) %>%
  dplyr::mutate(meanperprov = mean(value))%>%
  ungroup()


pM6_PC <- ggplot()+
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(prov), meanperprov),
                                    height = ..density.., 
                                    fill   = as.factor(max.Q.prov),
                                    vline_color = ..quantile..),
                                scale = 3, 
                                alpha = .6,
                                rel_min_height=c(.01),
                                size=0.3,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
  
  
  scale_y_discrete(labels=c("ALT"=parse(text = TeX("$cp_{ALT}$")),
                            "ARM"=parse(text = TeX("$cp_{ARM}$")),
                            "ARN"=parse(text = TeX("$cp_{ARN}$")),
                            "BAY"=parse(text = TeX("$cp_{BAY}$")),
                            "BON"=parse(text = TeX("$cp_{BON}$")),
                            "CAD"=parse(text = TeX("$cp_{CAD}$")),
                            "CAR"=parse(text = TeX("$cp_{CAR}$")),
                            "CAS"=parse(text = TeX("$cp_{CAS}$")),
                            "CEN"=parse(text = TeX("$cp_{CEN}$")),
                            "COC"=parse(text = TeX("$cp_{COC}$")),
                            "COM"=parse(text = TeX("$cp_{COM}$")),
                            "CUE"=parse(text = TeX("$cp_{CUE}$")),
                            "HOU"=parse(text = TeX("$cp_{HOU}$")),
                            "LAM"=parse(text = TeX("$cp_{LAM}$")),
                            "LEI"=parse(text = TeX("$cp_{LEI}$")),
                            "MAD"=parse(text = TeX("$cp_{MAD}$")),
                            "MIM"=parse(text = TeX("$cp_{MIM}$")),
                            "OLB"=parse(text = TeX("$cp_{OLB}$")),
                            "OLO"=parse(text = TeX("$cp_{OLO}$")),
                            "ORI"=parse(text = TeX("$cp_{ORI}$")),
                            "PET"=parse(text = TeX("$cp_{PET}$")),
                            "PIA"=parse(text = TeX("$cp_{PIA}$")),
                            "PIE"=parse(text = TeX("$cp_{PIE}$")),
                            "PLE"=parse(text = TeX("$cp_{PLE}$")),
                            "PUE"=parse(text = TeX("$cp_{PUE}$")),
                            "QUA"=parse(text = TeX("$cp_{QUA}$")),
                            "SAC"=parse(text = TeX("$cp_{SAC}$")),
                            "SAL"=parse(text = TeX("$cp_{SAL}$")),
                            "SEG"=parse(text = TeX("$cp_{SEG}$")),
                            "SIE"=parse(text = TeX("$cp_{SIE}$")),
                            "STJ"=parse(text = TeX("$cp_{STJ}$")),
                            "TAM"=parse(text = TeX("$cp_{TAM}$")),
                            "VAL"=parse(text = TeX("$cp_{VAL}$")),
                            "VER"=parse(text = TeX("$cp_{VER}$")))) +
  
   coord_cartesian(c(-0.35,0.25))+
  
   scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Median"),
                        name = NULL) +
  
  scale_fill_manual(values=c("orangered3",
                             "gold2",
                             "darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Northern Africa", 
                                                   "Corsica",
                                                   "Central Spain",
                                                   "French Atlantic",
                                                   "Iberian Atlantic",
                                                   "South-eastern Spain"),name="Gene pools") +
  labs(fill     = "Gene pools",
       y        = "", 
       x        = TeX("Provenance climate intercepts cp_{p}") ) + 
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))

```
 

## >> Fig. Prov, GenePool & ProvClim intercepts

In the manuscript.


```{r FigurePaperProvEffect, fig.width=12 ,fig.height=12,warning=F,message=F}
pM4_prov <- pM4_prov + theme(legend.position = "none")
pM4_GP <- pM4_GP + theme(legend.position = "none")

pM6_prov <- pM6_prov + theme(legend.position = "none")
pM6_GP <- pM6_GP + theme(legend.position = "none")
pM6_PC <- pM6_PC + theme(legend.position = "none")


g <-  ggarrange(ggarrange(pM3_prov,ggarrange(pM4_prov,pM4_GP,nrow=1),labels=c("A","B"),font.label = list(size = 20)),
                ggarrange(pM6_prov,pM6_GP,pM6_PC,nrow=1,labels=c("C"),font.label = list(size = 20)),nrow=3)


p3 <- plot_grid(pM3_prov) + border(color = "gray70",size=0.6)
p4 <- plot_grid(pM4_prov ,pM4_GP,nrow=1)+ border(color = "gray70",size=0.6)
p6 <- plot_grid(pM6_prov  ,pM6_GP,pM6_PC,nrow=1) + border(color = "gray70",size=0.6)

gg <-  ggarrange(ggarrange(p3,p4,nrow=1,labels=c("A","B"),font.label = list(size = 20)),
                 ggarrange(p6 ,labels=c("C"),font.label = list(size = 20)),nrow=2) 
gg  
```

```{r savePostDHPC}
ggsave(gg, file="../../figs/Extra/PostDHPC.png",width = 16,height=16)   
```




# **MODEL M7**

```{r loadM7}
mod <- readRDS(file="../../outputs/models/P1/MOD7.rds") 
```



## Main parameters

### Standard deviations

#### Mean

```{r TableM7SDMean, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_site__bio5_prov.sc'="$\\sigma_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma_{\\beta_{min.pre,s}}$",
                              'sd_site__gPEA.sc'="$\\sigma_{\\beta_{gPEA,s}}$",
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```

#### Median

```{r TableM7SDMedian, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_site__bio5_prov.sc'="$\\sigma_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma_{\\beta_{min.pre,s}}$",
                              'sd_site__gPEA.sc'="$\\sigma_{\\beta_{gPEA,s}}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs

```{r M7MCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            'sd_site__bio5_prov.sc'=parse(text = TeX("$\\sigma_{\\beta_{max.temp,s}}$")),
                            'sd_site__bio14_prov.sc'=parse(text = TeX("$\\sigma_{\\beta_{min.pre,s}}$")),
                            'sd_site__gPEA.sc'=parse(text = TeX("$\\sigma_{\\beta_{gPEA,s}}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_mmQ1Q2Q3Q4Q5Q6__Intercept"=parse(text = TeX("$\\sigma_{g_{j}}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM7VARMean, echo=F, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigm^{2}a_{G_{P}}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_site__bio5_prov.sc'="$\\sigma^{2}_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma^{2}_{\\beta_{min.pre,s}}$",
                              'sd_site__gPEA.sc'="$\\sigma^{2}_{\\beta_{gPEA,s}}$",
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM7VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_site__bio5_prov.sc'="$\\sigma^{2}_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma^{2}_{\\beta_{min.pre,s}}$",
                              'sd_site__gPEA.sc'="$\\sigma^{2}_{\\beta_{gPEA,s}}$",
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M7_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```




#### Figs

In the Supplementary Information.


```{r M7VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__b|sd_site__g|sd_site__c|sd_clon|sd_prov|sd_site:block|sigma|sd_mm)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_site__bio5_prov.sc)"=parse(text = TeX("$\\sigma^{2}_{\\beta_{max.temp,s}}$")),
                            "square(sd_site__bio14_prov.sc)"=parse(text = TeX("$\\sigma^{2}_{\\beta_{min.pre,s}}$")),
                            "square(sd_site__gPEA.sc)"=parse(text = TeX("$\\sigma^{2}_{\\beta_{gPEA,s}}$")),
                            "square(sd_prov:site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{Inter}$")),
                            "square(sd_mmQ1Q2Q3Q4Q5Q6__Intercept)"=parse(text = TeX("$\\sigma^{2}_{g_{j}}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__I)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(0.9,0.35,0.55),nrow=3,vjust=c(1.2,1,1))
ggsave(p,file="../../figs/SuppInfo/M7_PostMainVar.png",height=8) 
p
```



## Gene pool intercepts

### Table

```{r TableM7GenePoolIntercepts}
# gp <- str_c(rep("Q",6),1:6)
# par <-paste(paste0("'r_mmQ1Q2Q3Q4Q5Q6[",gp,",Intercept]'")) 
# par2 <-paste(paste0("'$g_{",1:6,"}$'")) 
# tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_mm)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_mmQ1Q2Q3Q4Q5Q6[Q1,Intercept]' = '$g_{NA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q2,Intercept]' = '$g_{C}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q3,Intercept]' = '$g_{CS}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q4,Intercept]' = '$g_{FA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q5,Intercept]' = '$g_{IA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q6,Intercept]' = '$g_{SES}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs 


#### MCMC intervals

```{r FigsM7GenePoolIntercepts,fig.height=3,warning=F,message=F}
gp <- str_c(rep("Q",6),1:6)
par <-paste(paste0("'r_mmQ1Q2Q3Q4Q5Q6[",gp,",Intercept]'")) 
par2 <-paste(paste0("'g_Q",1:6,"'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_mmQ1Q2Q3Q4Q5Q6",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +

  labs(title = "Posterior interval estimates from MCMC draws") +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_mmQ1Q2Q3Q4Q5Q6[Q1,Intercept]' = parse(text = TeX("$\\g_{NA}$")),
                            'r_mmQ1Q2Q3Q4Q5Q6[Q2,Intercept]' = parse(text = TeX("$\\g_{C}$")),
                            'r_mmQ1Q2Q3Q4Q5Q6[Q3,Intercept]' = parse(text = TeX("$\\g_{CS}$")),
                            'r_mmQ1Q2Q3Q4Q5Q6[Q4,Intercept]' = parse(text = TeX("$\\g_{FA}$")),
                            'r_mmQ1Q2Q3Q4Q5Q6[Q5,Intercept]' = parse(text = TeX("$\\g_{IA}$")),
                            'r_mmQ1Q2Q3Q4Q5Q6[Q6,Intercept]' = parse(text = TeX("$\\g_{SES}$"))))
```

#### Density ridges

```{r M7PostGenePoolIntercepts, fig.width=10,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_mmQ1Q2Q3Q4Q5Q6\\[")
colnames(POST) <- str_sub(colnames(POST),18,-12)
POST <- as.data.frame(t(POST))
POST$genepool <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -genepool)%>%
  group_by(genepool) %>%
  dplyr::mutate(meanpergenepool = mean(value))%>%
  ungroup()

pM4_GP <- ggplot()+
  geom_vline(xintercept = 0, 
             col        = "grey70") +
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(genepool), meanpergenepool),
                                    # height = ..density.., 
                                    fill   = as.factor(genepool),
                                    vline_color = ..quantile..),
                                scale = 2, 
                                alpha = .6,
                                rel_min_height=c(.001),
                                size=0.5,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
        scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Median"),
                        name = NULL) +
  coord_cartesian(c(-0.5,0.4))+
  scale_fill_manual(values=c("orangered3",
                             "gold2",
                             "darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Northern Africa", 
                                                   "Corsica",
                                                   "Central Spain",
                                                   "French Atlantic",
                                                   "Iberian Atlantic",
                                                   "South-eastern Spain")) +
  
  
  scale_y_discrete(labels=c("Q1"=parse(text = TeX("$g_{NA}$")),
                            "Q2"=parse(text = TeX("$g_{C}$")),
                            "Q3"=parse(text = TeX("$g_{CS}$")),
                            "Q4"=parse(text = TeX("$g_{FA}$")),
                            "Q5"=parse(text = TeX("$g_{IA}$")),
                            "Q6"=parse(text = TeX("$g_{SES}$")))) + 
  labs(fill     = "Gene pools",
       y        = "", 
       x        = TeX("Gene pool intercept g_{j}")
       ) + 
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
pM4_GP
```




## Maximum temperature of the warmest month

### Table

```{r M7BetaMaxTempTable, warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio5_prov.sc]'")) 
par2 <-paste(paste0("'$\\beta_{max.temp,",site,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^r_site.*bio5_prov.sc\\]")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 
                              'r_site[portugal,bio5_prov.sc]' = '$\\beta_{max.temp,Portugal}$',
                              'r_site[bordeaux,bio5_prov.sc]' = '$\\beta_{max.temp,Bordeaux}$',
                              'r_site[asturias,bio5_prov.sc]' = '$\\beta_{max.temp,Asturias}$',
                              'r_site[madrid,bio5_prov.sc]' = '$\\beta_{max.temp,Madrid}$',
                              'r_site[caceres,bio5_prov.sc]' = '$\\beta_{max.temp,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M7BetaMaxTempFigs,fig.width=8,fig.height=3,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio5_prov.sc]'")) 
par2 <-paste(paste0("'beta_MaxTemp_",str_to_title(site),"'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_site.*bio5_prov.sc\\]",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_site[portugal,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Portugal}$")),
                            'r_site[bordeaux,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Bordeaux}$")),
                            'r_site[asturias,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Asturias}$")),
                            'r_site[madrid,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Madrid}$")),
                            'r_site[caceres,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Caceres}$"))))
```


## Minimum precipitation of the driest month

### Table

```{r M7BetaMinPreTable, fig.height=5,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio14_prov.sc]'")) 
par2 <-paste(paste0("'$\\beta_{min.pre,",site,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^r_site.*bio14_prov.sc\\]")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_site[portugal,bio14_prov.sc]' = '$\\beta_{min.pre,Portugal}$',
                              'r_site[bordeaux,bio14_prov.sc]' = '$\\beta_{min.pre,Bordeaux}$',
                              'r_site[asturias,bio14_prov.sc]' = '$\\beta_{min.pre,Asturias}$',
                              'r_site[madrid,bio14_prov.sc]' = '$\\beta_{min.pre,Madrid}$',
                              'r_site[caceres,bio14_prov.sc]' = '$\\beta_{min.pre,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M7BetaMInPreFigs,fig.width=8,fig.height=3,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio14_prov.sc]'")) 
par2 <-paste(paste0("'beta_MinPre_",str_to_title(site),"'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_site.*bio14_prov.sc\\]",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_site[portugal,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Portugal}$")),
                            'r_site[bordeaux,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Bordeaux}$")),
                            'r_site[asturias,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Asturias}$")),
                            'r_site[madrid,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Madrid}$")),
                            'r_site[caceres,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Caceres}$"))))
```

## gPEAs

> = global Positive effect alleles

### Table

```{r M7BetagPEATable, warning=F,message=F}
# site <- unique(data$site)
# par <-paste(paste0("'r_site[",site,",gPEA.sc]'")) 
# par2 <-paste(paste0("'$\\beta_{gPEA,",site,"}$'")) 
# tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^r_site.*PEA.sc\\]")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_site[portugal,gPEA.sc]' = '$\\beta_{gPEA,Portugal}$',
                              'r_site[bordeaux,gPEA.sc]' = '$\\beta_{gPEA,Bordeaux}$',
                              'r_site[asturias,gPEA.sc]' = '$\\beta_{gPEA,Asturias}$',
                              'r_site[madrid,gPEA.sc]' = '$\\beta_{gPEA,Madrid}$',
                              'r_site[caceres,gPEA.sc]' = '$\\beta_{gPEA,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M7BetagPEAeFigs,fig.width=8,fig.height=3,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",gPEA.sc]'")) 
par2 <-paste(paste0("'beta_gPEA_",str_to_title(site),"'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_site.*PEA.sc\\]",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_site[portugal,gPEA.sc]' = parse(text = TeX("$\\beta_{gPEA,Portugal}$")),
                            'r_site[bordeaux,gPEA.sc]' = parse(text = TeX("$\\beta_{gPEA,Bordeaux}$")),
                            'r_site[asturias,gPEA.sc]' = parse(text = TeX("$\\beta_{gPEA,Asturias}$")),
                            'r_site[madrid,gPEA.sc]' = parse(text = TeX("$\\beta_{gPEA,Madrid}$")),
                            'r_site[caceres,gPEA.sc]' = parse(text = TeX("$\\beta_{gPEA,Caceres}$")))) 
```


# **MODEL M8**

```{r loadM8}
mod <- readRDS(file="../../outputs/models/P1/MOD8.rds")
```


## Main parameters

### Standard deviations

#### Mean

```{r TableM8SDMean, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_site__bio5_prov.sc'="$\\sigma_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma_{\\beta_{min.pre,s}}$",
                              'sd_site__rPEA.sc'="$\\sigma_{\\beta_{rPEA,s}}$",
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```

#### Median

```{r TableM8SDMedian, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_site__bio5_prov.sc'="$\\sigma_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma_{\\beta_{min.pre,s}}$",
                              'sd_site__rPEA.sc'="$\\sigma_{\\beta_{rPEA,s}}$",
                              'sd_prov:clon__Intercept'='$\\sigma_{G}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs


```{r M8MCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            'sd_site__bio5_prov.sc'=parse(text = TeX("$\\sigma_{\\beta_{max.temp,s}}$")),
                            'sd_site__bio14_prov.sc'=parse(text = TeX("$\\sigma_{\\beta_{min.pre,s}}$")),
                            'sd_site__rPEA.sc'=parse(text = TeX("$\\sigma_{\\beta_{rPEA,s}}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_mmQ1Q2Q3Q4Q5Q6__Intercept"=parse(text = TeX("$\\sigma_{g_{j}}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM8VARMean, echo=F, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigm^{2}a_{G_{P}}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_site__bio5_prov.sc'="$\\sigma^{2}_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma^{2}_{\\beta_{min.pre,s}}$",
                              'sd_site__rPEA.sc'="$\\sigma^{2}_{\\beta_{rPEA,s}}$",
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM8VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_site__bio5_prov.sc'="$\\sigma^{2}_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma^{2}_{\\beta_{min.pre,s}}$",
                              'sd_site__rPEA.sc'="$\\sigma^{2}_{\\beta_{rPEA,s}}$",
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M8_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```




#### Figs

In the Supplementary Information.


```{r M8VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__b|sd_site__r|sd_site__c|sd_clon|sd_prov|sd_site:block|sigma|sd_mm)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_site__bio5_prov.sc)"=parse(text = TeX("$\\sigma^{2}_{\\beta_{max.temp,s}}$")),
                            "square(sd_site__bio14_prov.sc)"=parse(text = TeX("$\\sigma^{2}_{\\beta_{min.pre,s}}$")),
                            "square(sd_site__rPEA.sc)"=parse(text = TeX("$\\sigma^{2}_{\\beta_{rPEA,s}}$")),
                            "square(sd_prov:site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{Inter}$")),
                            "square(sd_mmQ1Q2Q3Q4Q5Q6__Intercept)"=parse(text = TeX("$\\sigma^{2}_{g_{j}}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__I)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(0.9,0.35,0.55),nrow=3,vjust=c(1.2,1,1))
ggsave(p,file="../../figs/SuppInfo/M8_PostMainVar.png",height=8)  
p
```



## Gene pool intercepts

### Table

```{r TableM8GenePoolIntercepts}
# gp <- str_c(rep("Q",6),1:6)
# par <-paste(paste0("'r_mmQ1Q2Q3Q4Q5Q6[",gp,",Intercept]'")) 
# par2 <-paste(paste0("'$g_{",1:6,"}$'")) 
# tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_mm)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_mmQ1Q2Q3Q4Q5Q6[Q1,Intercept]' = '$g_{NA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q2,Intercept]' = '$g_{C}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q3,Intercept]' = '$g_{CS}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q4,Intercept]' = '$g_{FA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q5,Intercept]' = '$g_{IA}$',
                                    'r_mmQ1Q2Q3Q4Q5Q6[Q6,Intercept]' = '$g_{SES}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs 


#### MCMC intervals

```{r FigsM8GenePoolIntercepts,fig.height=3,warning=F,message=F}
gp <- str_c(rep("Q",6),1:6)
par <-paste(paste0("'r_mmQ1Q2Q3Q4Q5Q6[",gp,",Intercept]'")) 
par2 <-paste(paste0("'g_Q",1:6,"'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_mmQ1Q2Q3Q4Q5Q6",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +

  labs(title = "Posterior interval estimates from MCMC draws") +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_mmQ1Q2Q3Q4Q5Q6[Q1,Intercept]' = parse(text = TeX("$\\g_{NA}$")),
                            'r_mmQ1Q2Q3Q4Q5Q6[Q2,Intercept]' = parse(text = TeX("$\\g_{C}$")),
                            'r_mmQ1Q2Q3Q4Q5Q6[Q3,Intercept]' = parse(text = TeX("$\\g_{CS}$")),
                            'r_mmQ1Q2Q3Q4Q5Q6[Q4,Intercept]' = parse(text = TeX("$\\g_{FA}$")),
                            'r_mmQ1Q2Q3Q4Q5Q6[Q5,Intercept]' = parse(text = TeX("$\\g_{IA}$")),
                            'r_mmQ1Q2Q3Q4Q5Q6[Q6,Intercept]' = parse(text = TeX("$\\g_{SES}$"))))
```

#### Density ridges

```{r M8PostGenePoolIntercepts, fig.width=10,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_mmQ1Q2Q3Q4Q5Q6\\[")
colnames(POST) <- str_sub(colnames(POST),18,-12)
POST <- as.data.frame(t(POST))
POST$genepool <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -genepool)%>%
  group_by(genepool) %>%
  dplyr::mutate(meanpergenepool = mean(value))%>%
  ungroup()

pM4_GP <- ggplot()+
  geom_vline(xintercept = 0, 
             col        = "grey70") +
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(genepool), meanpergenepool),
                                    fill   = as.factor(genepool),
                                    vline_color = ..quantile..),
                                scale = 2, 
                                alpha = .6,
                                rel_min_height=c(.001),
                                size=0.5,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
        scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Median"),
                        name = NULL) +
  coord_cartesian(c(-0.5,0.4))+
  scale_fill_manual(values=c("orangered3",
                             "gold2",
                             "darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Northern Africa", 
                                                   "Corsica",
                                                   "Central Spain",
                                                   "French Atlantic",
                                                   "Iberian Atlantic",
                                                   "South-eastern Spain")) +
  
  
  scale_y_discrete(labels=c("Q1"=parse(text = TeX("$g_{NA}$")),
                            "Q2"=parse(text = TeX("$g_{C}$")),
                            "Q3"=parse(text = TeX("$g_{CS}$")),
                            "Q4"=parse(text = TeX("$g_{FA}$")),
                            "Q5"=parse(text = TeX("$g_{IA}$")),
                            "Q6"=parse(text = TeX("$g_{SES}$")))) + 
  labs(y        = "", 
       x        = TeX("Gene pool intercept g_{j}")) + 
  theme_bw() + theme(axis.text = element_text(size=12),axis.title = element_text(size=14), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))
pM4_GP
```




## Maximum temperature of the warmest month

### Table

```{r M8BetaMaxTempTable, warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio5_prov.sc]'")) 
par2 <-paste(paste0("'$\\beta_{max.temp,",site,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^r_site.*bio5_prov.sc\\]")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 
                              'r_site[portugal,bio5_prov.sc]' = '$\\beta_{max.temp,Portugal}$',
                              'r_site[bordeaux,bio5_prov.sc]' = '$\\beta_{max.temp,Bordeaux}$',
                              'r_site[asturias,bio5_prov.sc]' = '$\\beta_{max.temp,Asturias}$',
                              'r_site[madrid,bio5_prov.sc]' = '$\\beta_{max.temp,Madrid}$',
                              'r_site[caceres,bio5_prov.sc]' = '$\\beta_{max.temp,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M8BetaMaxTempFigs,fig.width=8,fig.height=3,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio5_prov.sc]'")) 
par2 <-paste(paste0("'beta_MaxTemp_",str_to_title(site),"'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_site.*bio5_prov.sc\\]",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_site[portugal,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Portugal}$")),
                            'r_site[bordeaux,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Bordeaux}$")),
                            'r_site[asturias,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Asturias}$")),
                            'r_site[madrid,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Madrid}$")),
                            'r_site[caceres,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Caceres}$"))))
```


## Minimum precipitation of the driest month

### Table

```{r M8BetaMinPreTable, fig.height=5,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio14_prov.sc]'")) 
par2 <-paste(paste0("'$\\beta_{min.pre,",site,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^r_site.*bio14_prov.sc\\]")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 'r_site[portugal,bio14_prov.sc]' = '$\\beta_{min.pre,Portugal}$',
                              'r_site[bordeaux,bio14_prov.sc]' = '$\\beta_{min.pre,Bordeaux}$',
                              'r_site[asturias,bio14_prov.sc]' = '$\\beta_{min.pre,Asturias}$',
                              'r_site[madrid,bio14_prov.sc]' = '$\\beta_{min.pre,Madrid}$',
                              'r_site[caceres,bio14_prov.sc]' = '$\\beta_{min.pre,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M8BetaMInPreFigs,fig.width=8,fig.height=3,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio14_prov.sc]'")) 
par2 <-paste(paste0("'beta_MinPre_",str_to_title(site),"'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_site.*bio14_prov.sc\\]",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_site[portugal,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Portugal}$")),
                            'r_site[bordeaux,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Bordeaux}$")),
                            'r_site[asturias,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Asturias}$")),
                            'r_site[madrid,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Madrid}$")),
                            'r_site[caceres,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Caceres}$"))))
```

## rPEAs

### Table

```{r M8BetarPEATable, warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",rPEA.sc]'")) 
par2 <-paste(paste0("'$\\beta_{rPEA,",site,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^r_site.*PEA.sc\\]")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 
                              'r_site[portugal,rPEA.sc]' = '$\\beta_{rPEA,Portugal}$',
                              'r_site[bordeaux,rPEA.sc]' = '$\\beta_{rPEA,Bordeaux}$',
                              'r_site[asturias,rPEA.sc]' = '$\\beta_{rPEA,Asturias}$',
                              'r_site[madrid,rPEA.sc]' = '$\\beta_{rPEA,Madrid}$',
                              'r_site[caceres,rPEA.sc]' = '$\\beta_{rPEA,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M8BetarPEAeFigs,fig.width=8,fig.height=3,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",rPEA.sc]'")) 
par2 <-paste(paste0("'beta_rPEA_",str_to_title(site),"'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_site.*PEA.sc\\]",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_site[portugal,rPEA.sc]' = parse(text = TeX("$\\beta_{rPEA,Portugal}$")),
                            'r_site[bordeaux,rPEA.sc]' = parse(text = TeX("$\\beta_{rPEA,Bordeaux}$")),
                            'r_site[asturias,rPEA.sc]' = parse(text = TeX("$\\beta_{rPEA,Asturias}$")),
                            'r_site[madrid,rPEA.sc]' = parse(text = TeX("$\\beta_{rPEA,Madrid}$")),
                            'r_site[caceres,rPEA.sc]' = parse(text = TeX("$\\beta_{rPEA,Caceres}$")))) 
```



# **Model M9**


```{r loadM9}
mod <- readRDS(file="../../outputs/models/P1/MOD9.rds")
```

## Main parameters

### Standard deviations

#### Mean

```{r TableM9SDMean, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```

#### Median

```{r TableM9SDMedian, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma_{g_{j}}$',
                              'sd_site:block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs


```{r M9MCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            "sd_site:block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "sd_mmQ1Q2Q3Q4Q5Q6__Intercept"=parse(text = TeX("$\\sigma_{g_{j}}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM9VARMean, echo=F, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigm^{2}a_{G_{P}}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM9VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_prov__Intercept'="$\\sigma^{2}_{P}$",
                              'sd_prov:clon__Intercept'='$\\sigma^{2}_{G}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_mmQ1Q2Q3Q4Q5Q6__Intercept'='$\\sigma^{2}_{g_{j}}$',
                              'sd_site:block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M9_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```


#### Figs

```{r M9VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__b|sd_site__c|sd_clon|sd_prov|sd_site:block|sigma|sd_mm)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_prov:clon__Intercept)"=parse(text = TeX("$\\sigma^{2}_{G}$")),
                            "square(sd_prov:site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{Inter}$")),
                            "square(sd_mmQ1Q2Q3Q4Q5Q6__Intercept)"=parse(text = TeX("$\\sigma^{2}_{g_{j}}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__I)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(0.9,0.35,0.55),nrow=3,vjust=c(1.2,1,1))
ggsave(p,file="../../figs/SuppInfo/M9_PostMainVar.png",height=8) 
p
```



## Gene pool intercepts

### Table

#### Mean

```{r TableM9GenePoolInterceptsMean, echo=F}
gp <- str_c(rep("Q",6),1:6)
par <-paste(paste0("'r_mmQ1Q2Q3Q4Q5Q6[",gp,",Intercept]'")) 
par2 <-paste(paste0("'$g_{",1:6,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^(r_mm)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 
                              'r_mmQ1Q2Q3Q4Q5Q6[Q1,Intercept]' = '$g_{NA}$',
                              'r_mmQ1Q2Q3Q4Q5Q6[Q2,Intercept]' = '$g_{C}$',
                              'r_mmQ1Q2Q3Q4Q5Q6[Q3,Intercept]' = '$g_{CS}$',
                              'r_mmQ1Q2Q3Q4Q5Q6[Q4,Intercept]' = '$g_{FA}$',
                              'r_mmQ1Q2Q3Q4Q5Q6[Q5,Intercept]' = '$g_{IA}$',
                              'r_mmQ1Q2Q3Q4Q5Q6[Q6,Intercept]' = '$g_{SES}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```




#### Median

```{r TableM9GenePoolInterceptsMedian, echo=F}
gp <- str_c(rep("Q",6),1:6)
par <-paste(paste0("'r_mmQ1Q2Q3Q4Q5Q6[",gp,",Intercept]'")) 
par2 <-paste(paste0("'$g_{",1:6,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "median",conf.int = T) %>% 
  filter(str_detect(term, "^(r_mm)")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 
                              'r_mmQ1Q2Q3Q4Q5Q6[Q1,Intercept]' = '$g_{NA}$',
                              'r_mmQ1Q2Q3Q4Q5Q6[Q2,Intercept]' = '$g_{C}$',
                              'r_mmQ1Q2Q3Q4Q5Q6[Q3,Intercept]' = '$g_{CS}$',
                              'r_mmQ1Q2Q3Q4Q5Q6[Q4,Intercept]' = '$g_{FA}$',
                              'r_mmQ1Q2Q3Q4Q5Q6[Q5,Intercept]' = '$g_{IA}$',
                              'r_mmQ1Q2Q3Q4Q5Q6[Q6,Intercept]' = '$g_{SES}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs 


```{r M9PostGenePoolIntercepts, fig.width=8,warning=F,message=F}
POST <- posterior_samples(mod,pars = "^r_mmQ1Q2Q3Q4Q5Q6\\[")
colnames(POST) <- str_sub(colnames(POST),18,-12)
POST <- as.data.frame(t(POST))
POST$genepool <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -genepool)%>%
  group_by(genepool) %>%
  dplyr::mutate(meanpergenepool = mean(value))%>%
  ungroup()

pGP <- ggplot()+
  geom_vline(xintercept = 0, 
             col        = "grey70") +
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = reorder(as.factor(genepool), meanpergenepool),
                                    fill   = as.factor(genepool),
                                    vline_color = ..quantile..),
                                scale = 2, 
                                alpha = .6,
                                rel_min_height=c(.001),
                                size=0.5,
                                quantile_lines = TRUE, quantiles = c(0.025,0.5,0.975)) +
        scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("2.5 and 97.5th percentiles", "Median"),
                        name = NULL) +
  coord_cartesian(c(-0.5,0.4))+
  scale_fill_manual(values=c("orangered3",
                             "gold2",
                             "darkorchid3",
                             "navyblue",
                             "turquoise2",
                             "green3"), labels = c("Northern Africa", 
                                                   "Corsica",
                                                   "Central Spain",
                                                   "French Atlantic",
                                                   "Iberian Atlantic",
                                                   "South-eastern Spain")) +
  scale_y_discrete(labels=c("Q1"=parse(text = TeX("$g_{NA}$")),
                            "Q2"=parse(text = TeX("$g_{C}$")),
                            "Q3"=parse(text = TeX("$g_{CS}$")),
                            "Q4"=parse(text = TeX("$g_{FA}$")),
                            "Q5"=parse(text = TeX("$g_{IA}$")),
                            "Q6"=parse(text = TeX("$g_{SES}$")))) + 
  labs(fill     = "Gene pools",
       y        = "", 
       x        = TeX("Gene pool intercept g_{j}")
       ) + 
  theme_bw() + theme(axis.text = element_text(size=18),axis.title = element_text(size=18), 
                        legend.text = element_text(size=18),legend.title = element_text(size=20))

pGP
ggsave(pGP, file="../../figs/SuppInfo/M9_GenePoolIntercepts.png",width = 16,height=10)
```



# **MODEL M10**

```{r loadM10}
mod <- readRDS(file="../../outputs/models/P1/MOD10.rds")
```



## Main parameters

### Standard deviations

#### Mean

```{r TableM10SDMean, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_site__bio5_prov.sc'="$\\sigma_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma_{\\beta_{min.pre,s}}$",
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```

#### Median

```{r TableM10SDMedian, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),
                              'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_site__bio5_prov.sc'="$\\sigma_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma_{\\beta_{min.pre,s}}$",
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs

```{r M10MCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            'sd_site__bio5_prov.sc'=parse(text = TeX("$\\sigma_{\\beta_{max.temp,s}}$")),
                            'sd_site__bio14_prov.sc'=parse(text = TeX("$\\sigma_{\\beta_{min.pre,s}}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM10VARMean, echo=F, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_site__bio5_prov.sc'="$\\sigma^{2}_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma^{2}_{\\beta_{min.pre,s}}$",
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM10VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site__bio5_prov.sc'="$\\sigma^{2}_{\\beta_{max.temp,s}}$",
                              'sd_site__bio14_prov.sc'="$\\sigma^{2}_{\\beta_{min.pre,s}}$",
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M10_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```




#### Figs

In the Supplementary Information.


```{r M10VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__b|sd_site__c|sd_clon|sd_prov|sd_site:block|sigma|sd_mm)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__bio5_prov.sc)"=parse(text = TeX("$\\sigma^{2}_{\\beta_{max.temp,s}}$")),
                            "square(sd_site__bio14_prov.sc)"=parse(text = TeX("$\\sigma^{2}_{\\beta_{min.pre,s}}$")),
                             "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__I)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(0.9,0.35,0.55),nrow=3,vjust=c(1.2,1,1))
ggsave(p,file="../../figs/SuppInfo/M10_PostMainVar.png",height=8) 
p
```


## Maximum temperature of the warmest month

### Table

```{r M10BetaMaxTempTable, fig.height=5,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio5_prov.sc]'")) 
par2 <-paste(paste0("'$\\beta_{max.temp,",site,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^r_site.*bio5_prov.sc\\]")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 
                              'r_site[portugal,bio5_prov.sc]' = '$\\beta_{max.temp,Portugal}$',
                              'r_site[bordeaux,bio5_prov.sc]' = '$\\beta_{max.temp,Bordeaux}$',
                              'r_site[asturias,bio5_prov.sc]' = '$\\beta_{max.temp,Asturias}$',
                              'r_site[madrid,bio5_prov.sc]' = '$\\beta_{max.temp,Madrid}$',
                              'r_site[caceres,bio5_prov.sc]' = '$\\beta_{max.temp,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M10BetaMaxTempFigs,fig.width=8,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio5_prov.sc]'")) 
par2 <-paste(paste0("'beta_MaxTemp_",str_to_title(site),"'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_site.*bio5_prov.sc\\]",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_site[portugal,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Portugal}$")),
                            'r_site[bordeaux,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Bordeaux}$")),
                            'r_site[asturias,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Asturias}$")),
                            'r_site[madrid,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Madrid}$")),
                            'r_site[caceres,bio5_prov.sc]' = parse(text = TeX("$\\beta_{max.temp,Caceres}$"))))
```


## Minimum precipitation of the driest month

### Table

```{r M10BetaMinPreTable, fig.height=5,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio14_prov.sc]'")) 
par2 <-paste(paste0("'$\\beta_{min.pre,",site,"}$'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^r_site.*bio14_prov.sc\\]")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 
                              'r_site[portugal,bio14_prov.sc]' = '$\\beta_{min.pre,Portugal}$',
                              'r_site[bordeaux,bio14_prov.sc]' = '$\\beta_{min.pre,Bordeaux}$',
                              'r_site[asturias,bio14_prov.sc]' = '$\\beta_{min.pre,Asturias}$',
                              'r_site[madrid,bio14_prov.sc]' = '$\\beta_{min.pre,Madrid}$',
                              'r_site[caceres,bio14_prov.sc]' = '$\\beta_{min.pre,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M10BetaMInPreFigs,fig.width=8,fig.height=3,warning=F,message=F}
site <- unique(data$site)
par <-paste(paste0("'r_site[",site,",bio14_prov.sc]'")) 
par2 <-paste(paste0("'beta_MinPre_",str_to_title(site),"'")) 
tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_site.*bio14_prov.sc\\]",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_site[portugal,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Portugal}$")),
                            'r_site[bordeaux,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Bordeaux}$")),
                            'r_site[asturias,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Asturias}$")),
                            'r_site[madrid,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Madrid}$")),
                            'r_site[caceres,bio14_prov.sc]' = parse(text = TeX("$\\beta_{min.pre,Caceres}$"))))
```

### Both

```{r figsbetaM10, fig.width=20,fig.height=8,message=F,warning=F}
POST <- posterior_samples(mod,pars = "^r_site\\[.*sc\\]") %>% dplyr::rename(
  beta_MinPre_Portugal = 'r_site[portugal,bio14_prov.sc]',
  beta_MinPre_Bordeaux = 'r_site[bordeaux,bio14_prov.sc]',
  beta_MinPre_Asturias = 'r_site[asturias,bio14_prov.sc]',
  beta_MinPre_Madrid = 'r_site[madrid,bio14_prov.sc]',
  beta_MinPre_Caceres = 'r_site[caceres,bio14_prov.sc]',
  beta_MaxTemp_Portugal = 'r_site[portugal,bio5_prov.sc]',
  beta_MaxTemp_Bordeaux = 'r_site[bordeaux,bio5_prov.sc]',
  beta_MaxTemp_Asturias = 'r_site[asturias,bio5_prov.sc]',
  beta_MaxTemp_Madrid = 'r_site[madrid,bio5_prov.sc]',
  beta_MaxTemp_Caceres = 'r_site[caceres,bio5_prov.sc]'
  )
POST <- as.data.frame(t(POST))
POST$var <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -var)


figs.beta <- ggplot()+
  geom_vline(xintercept = 0, col="grey70") +
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = var,
                                    fill   = as.factor(var),
                                    vline_color = ..quantile..),
                                scale = 1.8, 
                                alpha = .8,
                                size=0.8,
                                rel_min_height=.01,
                                quantile_lines = TRUE, 
                                quantiles = c(0.025,0.5,0.975)) +
  # coord_cartesian(c(-,0.8))+
      scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("5% & 95% quantiles", "mean"),
                        name = NULL) +
  scale_y_discrete(labels=c("beta_MaxTemp_Caceres"=parse(text = TeX("$\\beta_{max.temp,Caceres}$")),
                            "beta_MaxTemp_Bordeaux"=parse(text = TeX("$\\beta_{max.temp,Bordeaux}$")),
                            "beta_MaxTemp_Portugal"=parse(text = TeX("$\\beta_{max.temp,Portugal}$")),
                            "beta_MaxTemp_Madrid"=parse(text = TeX("$\\beta_{max.temp,Madrid}$")),
                            "beta_MaxTemp_Asturias"=parse(text = TeX("$\\beta_{max.temp,Asturias}$")),
                            "beta_MinPre_Caceres"=parse(text = TeX("$\\beta_{min.pre,Caceres}$")),
                            "beta_MinPre_Bordeaux"=parse(text = TeX("$\\beta_{min.pre,Bordeaux}$")),
                            "beta_MinPre_Portugal"=parse(text = TeX("$\\beta_{min.pre,Portugal}$")),
                            "beta_MinPre_Madrid"=parse(text = TeX("$\\beta_{min.pre,Madrid}$")),
                            "beta_MinPre_Asturias"=parse(text = TeX("$\\beta_{min.pre,Asturias}$")))) + 
  labs(title="",
       y        = "", 
       x = "Parameter estimates"
       #x        = TeX(paste0("Estimate of the varying slopes $\\beta_{",pea,",s}$, $\\beta_{min.pre,s}$ and $\\beta_{max.temp,s}$"))
       ) + 
  scale_fill_manual(values=c(vir_lite("cyan2",ds=ds),
                             vir_lite("navyblue",ds=ds),
                             vir_lite("pink",ds=ds),
                             vir_lite("deeppink",ds=ds),
                             vir_lite("dodgerblue2",ds=ds),
                             vir_lite("cyan2",ds=ds),
                             vir_lite("navyblue",ds=ds),
                             vir_lite("pink",ds=ds),
                             vir_lite("deeppink",ds=ds),
                             vir_lite("dodgerblue2",ds=ds))) +
  theme_bw() + theme(axis.text = element_text(size=22),axis.title = element_text(size=22), 
                    legend.position = "none", plot.title = element_text(size=22)) 

figs.beta
ggsave(figs.beta, file="../../figs/SuppInfo/M10_ProvClimateVariablesIntercepts.png",width = 16,height=12)
```



# **Model M11**


```{r loadM11}
mod <- readRDS(file="../../outputs/models/P1/MOD11.rds")
```



## Main parameters

### Standard deviations

#### Mean

```{r TableM11SDMean, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_site__gPEA.sc'="$\\sigma_{\\beta_{gPEA,s}}$",
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```

#### Median

```{r TableM11SDMedian, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_site__gPEA.sc'="$\\sigma_{\\beta_{gPEA,s}}$",
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs


```{r M11MCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            'sd_site__gPEA.sc'=parse(text = TeX("$\\sigma_{\\beta_{gPEA,s}}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM11VARMean, echo=F, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_site__gPEA.sc'="$\\sigma^{2}_{\\beta_{gPEA,s}}$",
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM11VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site__gPEA.sc'="$\\sigma^{2}_{\\beta_{gPEA,s}}$",
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M11_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```




#### Figs

In the Supplementary Information.


```{r M11VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__g|sd_site__c|sd_clon|sd_prov|sd_site:block|sigma|sd_mm)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__gPEA.sc)"=parse(text = TeX("$\\sigma^{2}_{\\beta_{gPEA,s}}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__I)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(0.9,0.35,0.55),nrow=3,vjust=c(1.2,1,1))
ggsave(p,file="../../figs/SuppInfo/M11_PostMainVar.png",height=6) 
p
```


## gPEAs

> = global Positive effect alleles

### Table

```{r M11BetagPEATable, fig.height=5,warning=F,message=F}
# site <- unique(data$site)
# par <-paste(paste0("'r_site[",site,",gPEA.sc]'")) 
# par2 <-paste(paste0("'$\\beta_{gPEA,",site,"}$'")) 
# tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^r_site.*all.sc\\]")) %>% 
  rename_all(str_to_title) %>% 
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 
                              'r_site[portugal,gPEA.sc]' = '$\\beta_{gPEA,Portugal}$',
                              'r_site[bordeaux,gPEA.sc]' = '$\\beta_{gPEA,Bordeaux}$',
                              'r_site[asturias,gPEA.sc]' = '$\\beta_{gPEA,Asturias}$',
                              'r_site[madrid,gPEA.sc]' = '$\\beta_{gPEA,Madrid}$',
                              'r_site[caceres,gPEA.sc]' = '$\\beta_{gPEA,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M11BetagPEAeFigs,fig.width=8,warning=F,message=F}
# site <- unique(data$site)
# par <-paste(paste0("'r_site[",site,",gPEA.sc]'")) 
# par2 <-paste(paste0("'beta_gPEA_",str_to_title(site),"'")) 
# tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_site.*PEA.sc\\]",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_site[portugal,gPEA.sc]' = parse(text = TeX("$\\beta_{gPEA,Portugal}$")),
                            'r_site[bordeaux,gPEA.sc]' = parse(text = TeX("$\\beta_{gPEA,Bordeaux}$")),
                            'r_site[asturias,gPEA.sc]' = parse(text = TeX("$\\beta_{gPEA,Asturias}$")),
                            'r_site[madrid,gPEA.sc]' = parse(text = TeX("$\\beta_{gPEA,Madrid}$")),
                            'r_site[caceres,gPEA.sc]' = parse(text = TeX("$\\beta_{gPEA,Caceres}$")))) 
```



```{r figsbetaM11, fig.width=8,message=F,warning=F}
pea="gPEA"
variable ="gPEA.sc"
  
POST <- posterior_samples(mod,pars = "^r_site\\[.*sc\\]") %>% dplyr::rename(
  beta_PEA_Portugal = paste0('r_site[portugal,',variable,']'),
  beta_PEA_Bordeaux = paste0('r_site[bordeaux,',variable,']'),
  beta_PEA_Asturias = paste0('r_site[asturias,',variable,']'),
  beta_PEA_Madrid = paste0('r_site[madrid,',variable,']'),
  beta_PEA_Caceres = paste0('r_site[caceres,',variable,']')
  )
POST <- as.data.frame(t(POST))
POST$var <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -var)


figs.beta <- ggplot()+
  geom_vline(xintercept = 0, col="grey70") +
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = var,
                                    fill   = as.factor(var),
                                    vline_color = ..quantile..),
                                scale = 1.8, 
                                alpha = .8,
                                size=0.8,
                                rel_min_height=.01,
                                quantile_lines = TRUE, 
                                quantiles = c(0.025,0.5,0.975)) +
  # coord_cartesian(c(-,0.8))+
      scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("5% & 95% quantiles", "mean"),
                        name = NULL) +
  scale_y_discrete(labels=c("beta_PEA_Caceres"=parse(text = TeX(paste0("$\\beta_{",pea,",Caceres}$"))),
                            "beta_PEA_Madrid"=parse(text = TeX(paste0("$\\beta_{",pea,",Madrid}$"))),
                            "beta_PEA_Portugal"=parse(text = TeX(paste0("$\\beta_{",pea,",Portugal}$"))),
                            "beta_PEA_Asturias"=parse(text = TeX(paste0("$\\beta_{",pea,",Asturias}$"))),
                            "beta_PEA_Bordeaux"=parse(text = TeX(paste0("$\\beta_{",pea,",Bordeaux}$")))
                            )) + 
  labs(y        = "", 
       x = TeX(paste0("Estimate of parameters $\\beta_{",pea,",s}$"))) + 
  scale_fill_manual(values=c(vir_lite("cyan2",ds=ds),
                             vir_lite("navyblue",ds=ds),
                             vir_lite("pink",ds=ds),
                             vir_lite("deeppink",ds=ds),
                             vir_lite("dodgerblue2",ds=ds))) +
  theme_bw() + theme(axis.text = element_text(size=22),axis.title = element_text(size=22), 
                    legend.position = "none", plot.title = element_text(size=22)) 
figs.beta
ggsave(figs.beta, file="../../figs/SuppInfo/M11_gpeaIntercepts.png",width = 16,height=12)   
```





# **Model M12**


```{r loadM12}
mod <- readRDS(file="../../outputs/models/P1/MOD12.rds") 
```


## Main parameters

### Standard deviations

#### Mean

```{r TableM12SDMean, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_site__rPEA.sc'="$\\sigma_{\\beta_{rPEA,s}}$",
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```

#### Median

```{r TableM12SDMedian, echo=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b|sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)
df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))

for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)
}

df %>% mutate(Parameter = recode_factor(rownames(df),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'sigma'='$\\sigma$',
                              'b_Intercept'='$\\beta_{0}$',
                              'sd_prov__Intercept'="$\\sigma_{P}$",
                              'sd_site__rPEA.sc'="$\\sigma_{\\beta_{rPEA,s}}$",
                              'sd_site__Intercept'='$\\sigma_{S}$',
                              'sd_block__Intercept'='$\\sigma_{B}$')) %>% 
  dplyr::select(Parameter,Median,SD,InfCI,SupCI) %>%
  remove_rownames() %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```


#### Figs


```{r M12MCMCIntervals, fig.width=10,fig.height=3,warning=F,message=F}
mod %>%  mcmc_intervals(regex_pars = "^(sd|sigma|b)",
                    prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() + 
  scale_y_discrete(labels=c("sigma"=parse(text=TeX("$\\sigma$")),
                            'b_Intercept'=parse(text = TeX("$\\beta_{0}$")),
                            "sd_site__Intercept"=parse(text = TeX("$\\sigma_{S}$")),
                            'sd_site__rPEA.sc'=parse(text = TeX("$\\sigma_{\\beta_{rPEA,s}}$")),
                            "sd_block__Intercept"=parse(text = TeX("$\\sigma_{B}$")),
                            "b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")))) +
  theme(axis.text = element_text(size=16))
```


### Variances

#### Mean 


```{r TableM12VARMean, echo=F, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Mean"] <- mean(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),
                              'sigma'='$\\sigma^{2}$',
                              'sd_site__rPEA.sc'="$\\sigma^{2}_{\\beta_{rPEA,s}}$",
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Mean","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Mean"] <- mean(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Mean,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")
```



#### Median

In the Supplementary Information.

```{r TableM12VARMedian, message=F, warning=F}
pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(sd|sigma)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- square(sims[, , i])
  quan <- unname(quantile(sims_i, probs = probs))
  df[i,"InfCI"] <- quan[1]
  df[i,"SupCI"] <- quan[2]
  df[i,"Median"] <- median(sims_i)
  df[i,"SD"] <- sd(sims_i)}

df <- df %>% mutate(Parameter = recode_factor(rownames(df),'sigma'='$\\sigma^{2}$',
                              'sd_site__Intercept'='$\\sigma^{2}_{S}$',
                              'sd_site__rPEA.sc'="$\\sigma^{2}_{\\beta_{rPEA,s}}$",
                              'sd_block__Intercept'='$\\sigma^{2}_{B}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


pars <- mod %>% get_variables() %>%  as.vector() %>% str_subset("^(b)")
sims <- as.array(mod, pars = pars, fixed = TRUE)

df2 <- as.data.frame(matrix(NA,length(pars),4,dimnames = list(pars,c("Median","SD","InfCI","SupCI"))))
for(i in pars){
  sims_i <- sims[, , i]
  quan <- unname(quantile(sims_i, probs = probs))
  df2[i,"InfCI"] <- quan[1]
  df2[i,"SupCI"] <- quan[2]
  df2[i,"Median"] <- median(sims_i)
  df2[i,"SD"] <- sd(sims_i)}
df2 <- df2 %>% mutate(Parameter = recode_factor(rownames(df2),'b_age.sc'="$\\beta_{age}$",
                              'b_Iage.scE2'= '$\\beta_{age2}$',
                              'b_Intercept'='$\\beta_{0}$')) %>% 
  remove_rownames() %>%
  dplyr::select(Parameter,Median,SD,InfCI,SupCI)


bind_rows(df,df2) %>% 
  knitr::kable(digits = 3 ,format = "markdown")

tab <- bind_rows(df,df2)
print(xtable(tab, type = "latex",digits=3), file = paste0("../../tables/Posteriors/M12_MainVarPost.tex"), include.rownames=FALSE,sanitize.text.function = function(x) {x})
```




#### Figs

In the Supplementary Information.


```{r M12VARFigs,fig.height=6, fig.width=10,warning=F,message=F}
p1 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__r|sd_site__c|sd_clon|sd_prov|sd_site:block|sigma|sd_mm)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__rPEA.sc)"=parse(text = TeX("$\\sigma^{2}_{\\beta_{rPEA,s}}$")),
                            "square(sd_site:block__Intercept)"=parse(text = TeX("$\\sigma^{2}_{B}$")),
                            "square(sigma)"=parse(text = TeX("$\\sigma^{2}$"))
                            )) 

p2 <- mod %>%  mcmc_intervals(regex_pars = "^(sd_site__I)",transformations="square",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("square(sd_site__Intercept)"=parse(text = TeX("$\\sigma^{2}_{S}$")))) 

p3 <- mod %>%  mcmc_intervals(regex_pars = "^(b_)",
                     prob=prob,prob_outer=prob_outer,point_est = "median") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c("b_age.sc"=parse(text = TeX("$\\beta_{age}$")),
                            "b_Iage.scE2"=parse(text = TeX("$\\beta_{age2}$")),
                            "b_Intercept"=parse(text = TeX("$\\beta_{0}$")))) 

p <- ggarrange(p1,p2,p3,labels=c("A","B","C"),font.label = list(size=20),heights = c(0.9,0.35,0.55),nrow=3,vjust=c(1.2,1,1))
ggsave(p,file="../../figs/SuppInfo/M12_PostMainVar.png",height=6)  
p
```




## rPEAs

> = region-specific Positive effect alleles

### Table

```{r M12BetagPEATable, fig.height=5,warning=F,message=F}
# site <- unique(data$site)
# par <-paste(paste0("'r_site[",site,",rPEA.sc]'")) 
# par2 <-paste(paste0("'$\\beta_{rPEA,",site,"}$'")) 
# tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

df <- mod %>% broom::tidyMCMC(estimate.method = "mean",conf.int = T) %>% 
  filter(str_detect(term, "^r_site.*PEA.sc\\]")) %>% 
  rename_all(str_to_title) %>%  
  dplyr::rename("Mean"=Estimate) %>% 
  mutate(Term = recode_factor(Term, 
                              'r_site[portugal,rPEA.sc]' = '$\\beta_{rPEA,Portugal}$',
                              'r_site[bordeaux,rPEA.sc]' = '$\\beta_{rPEA,Bordeaux}$',
                              'r_site[asturias,rPEA.sc]' = '$\\beta_{rPEA,Asturias}$',
                              'r_site[madrid,rPEA.sc]' = '$\\beta_{rPEA,Madrid}$',
                              'r_site[caceres,rPEA.sc]' = '$\\beta_{rPEA,Caceres}$'))
df %>% knitr::kable(digits = 3 ,format = "markdown")
```

### Figs

```{r M12BetagPEAeFigs,fig.width=8,fig.height=3,warning=F,message=F}
# site <- unique(data$site)
# par <-paste(paste0("'r_site[",site,",rPEA.sc]'")) 
# par2 <-paste(paste0("'beta_rPEA_",str_to_title(site),"'")) 
# tocopy <- noquote(str_sub(stri_paste(paste0(par," = ",par2,","),collapse = ""),0,-2))

mod %>%  mcmc_intervals(regex_pars = "^r_site.*PEA.sc\\]",
                     prob=prob,prob_outer=prob_outer,point_est = "mean") +  theme_bw() +
  theme(axis.text = element_text(size=16)) +
  scale_y_discrete(labels=c('r_site[portugal,rPEA.sc]' = parse(text = TeX("$\\beta_{rPEA,Portugal}$")),
                            'r_site[bordeaux,rPEA.sc]' = parse(text = TeX("$\\beta_{rPEA,Bordeaux}$")),
                            'r_site[asturias,rPEA.sc]' = parse(text = TeX("$\\beta_{rPEA,Asturias}$")),
                            'r_site[madrid,rPEA.sc]' = parse(text = TeX("$\\beta_{rPEA,Madrid}$")),
                            'r_site[caceres,rPEA.sc]' = parse(text = TeX("$\\beta_{rPEA,Caceres}$")))) 
```


```{r figsbetaM12, fig.width=20,fig.height=6,message=F,warning=F}
pea="rPEA"
variable ="rPEA.sc"
  
POST <- posterior_samples(mod,pars = "^r_site\\[.*sc\\]") %>% dplyr::rename(
  beta_PEA_Portugal = paste0('r_site[portugal,',variable,']'),
  beta_PEA_Bordeaux = paste0('r_site[bordeaux,',variable,']'),
  beta_PEA_Asturias = paste0('r_site[asturias,',variable,']'), 
  beta_PEA_Madrid = paste0('r_site[madrid,',variable,']'),
  beta_PEA_Caceres = paste0('r_site[caceres,',variable,']')
  )
POST <- as.data.frame(t(POST))
POST$var <- as.factor(rownames(POST))

posteriorsimpelmodellong <- POST %>%  as_tibble() %>% 
gather(key = "key", value = "value", -var)


figs.beta <- ggplot()+
  geom_vline(xintercept = 0, col="grey70") +
  stat_density_ridges(data  = posteriorsimpelmodellong, 
                                aes(x      = value,
                                    y      = var,
                                    fill   = as.factor(var),
                                    vline_color = ..quantile..),
                                scale = 1.8, 
                                alpha = .8,
                                size=0.8,
                                rel_min_height=.01,
                                quantile_lines = TRUE, 
                                quantiles = c(0.025,0.5,0.975)) +
  # coord_cartesian(c(-,0.8))+
      scale_discrete_manual("vline_color",
                        values = c("blue", "red", "blue", "black"), 
                        breaks = c(1, 2),
                        labels = c("5% & 95% quantiles", "mean"),
                        name = NULL) +
  scale_y_discrete(labels=c("beta_PEA_Caceres"=parse(text = TeX(paste0("$\\beta_{",pea,",Caceres}$"))),
                            "beta_PEA_Madrid"=parse(text = TeX(paste0("$\\beta_{",pea,",Madrid}$"))),
                            "beta_PEA_Portugal"=parse(text = TeX(paste0("$\\beta_{",pea,",Portugal}$"))),
                            "beta_PEA_Asturias"=parse(text = TeX(paste0("$\\beta_{",pea,",Asturias}$"))),
                            "beta_PEA_Bordeaux"=parse(text = TeX(paste0("$\\beta_{",pea,",Bordeaux}$")))
                            )) + 
  labs(title="",
       y        = "", 
       x = TeX(paste0("Estimate of parameters $\\beta_{",pea,",s}$"))
       ) + 
  scale_fill_manual(values=c(vir_lite("cyan2",ds=ds),
                             vir_lite("navyblue",ds=ds),
                             vir_lite("pink",ds=ds),
                             vir_lite("deeppink",ds=ds),
                             vir_lite("dodgerblue2",ds=ds))) +
  theme_bw() + theme(axis.text = element_text(size=22),axis.title = element_text(size=22), 
                    legend.position = "none", plot.title = element_text(size=22)) 
figs.beta
ggsave(figs.beta, file="../../figs/SuppInfo/M12_rpeaIntercepts.png",width = 16,height=12)  
```



`r knitr::opts_chunk$set(eval = F)`


